{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse the results of new clixo, output relevant files for cytoscape analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import networkx as nx\n",
    "import pandas as pd\n",
    "import random\n",
    "\n",
    "\n",
    "# latex rendering of text in graphs\n",
    "import matplotlib as mpl\n",
    "\n",
    "from matplotlib import rcParams\n",
    "rcParams['font.family'] = 'sans-serif'\n",
    "rcParams['font.sans-serif'] = ['Arial']\n",
    "\n",
    "sns.set_style('white')\n",
    "\n",
    "import sys\n",
    "from IPython.display import display\n",
    "% matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ddot import Ontology"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load the network genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "844\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>dual_phenotype_snv_indel_DECIPHERv9.25</th>\n",
       "      <th>dual_phenotype_snv_indel_DECIPHERv9.31</th>\n",
       "      <th>dual_phenotype_small_CNV_less1MB_DECIPHERv9.25</th>\n",
       "      <th>PCGC/PHN dual phenotype dDNV</th>\n",
       "      <th>sum_damaging_variants (DECIPHER 9.25, PCGC/PHN)</th>\n",
       "      <th>ASD_HC</th>\n",
       "      <th>CHD_HC</th>\n",
       "      <th>z_ASD</th>\n",
       "      <th>z_CHD</th>\n",
       "      <th>z_both</th>\n",
       "      <th>gtex_heart_percentile_exp</th>\n",
       "      <th>gtex_brain_percentile_exp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NAA15</th>\n",
       "      <td>NAA15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>18.901197</td>\n",
       "      <td>20.892283</td>\n",
       "      <td>394.889167</td>\n",
       "      <td>0.733689</td>\n",
       "      <td>0.620830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>16.395135</td>\n",
       "      <td>18.819067</td>\n",
       "      <td>308.541142</td>\n",
       "      <td>0.896062</td>\n",
       "      <td>0.894637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POGZ</th>\n",
       "      <td>POGZ</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14.186540</td>\n",
       "      <td>14.641398</td>\n",
       "      <td>207.710787</td>\n",
       "      <td>0.908309</td>\n",
       "      <td>0.953178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KMT2C</th>\n",
       "      <td>KMT2C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>13.462226</td>\n",
       "      <td>14.954173</td>\n",
       "      <td>201.316464</td>\n",
       "      <td>0.873100</td>\n",
       "      <td>0.908995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM5B</th>\n",
       "      <td>KDM5B</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>10.464431</td>\n",
       "      <td>13.490241</td>\n",
       "      <td>141.167693</td>\n",
       "      <td>0.677893</td>\n",
       "      <td>0.698533</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  dual_phenotype_snv_indel_DECIPHERv9.25  \\\n",
       "gene                                                   \n",
       "NAA15  NAA15                                     0.0   \n",
       "PTEN    PTEN                                     1.0   \n",
       "POGZ    POGZ                                     3.0   \n",
       "KMT2C  KMT2C                                     0.0   \n",
       "KDM5B  KDM5B                                     1.0   \n",
       "\n",
       "       dual_phenotype_snv_indel_DECIPHERv9.31  \\\n",
       "gene                                            \n",
       "NAA15                                     0.0   \n",
       "PTEN                                      1.0   \n",
       "POGZ                                      3.0   \n",
       "KMT2C                                     0.0   \n",
       "KDM5B                                     1.0   \n",
       "\n",
       "       dual_phenotype_small_CNV_less1MB_DECIPHERv9.25  \\\n",
       "gene                                                    \n",
       "NAA15                                             0.0   \n",
       "PTEN                                              0.0   \n",
       "POGZ                                              0.0   \n",
       "KMT2C                                             0.0   \n",
       "KDM5B                                             0.0   \n",
       "\n",
       "       PCGC/PHN dual phenotype dDNV  \\\n",
       "gene                                  \n",
       "NAA15                           0.0   \n",
       "PTEN                            1.0   \n",
       "POGZ                            1.0   \n",
       "KMT2C                           1.0   \n",
       "KDM5B                           1.0   \n",
       "\n",
       "       sum_damaging_variants (DECIPHER 9.25, PCGC/PHN)  ASD_HC  CHD_HC  \\\n",
       "gene                                                                     \n",
       "NAA15                                              0.0       1       1   \n",
       "PTEN                                               2.0       1       1   \n",
       "POGZ                                               4.0       1       1   \n",
       "KMT2C                                              1.0       1       1   \n",
       "KDM5B                                              2.0       1       1   \n",
       "\n",
       "           z_ASD      z_CHD      z_both  gtex_heart_percentile_exp  \\\n",
       "gene                                                                 \n",
       "NAA15  18.901197  20.892283  394.889167                   0.733689   \n",
       "PTEN   16.395135  18.819067  308.541142                   0.896062   \n",
       "POGZ   14.186540  14.641398  207.710787                   0.908309   \n",
       "KMT2C  13.462226  14.954173  201.316464                   0.873100   \n",
       "KDM5B  10.464431  13.490241  141.167693                   0.677893   \n",
       "\n",
       "       gtex_brain_percentile_exp  \n",
       "gene                              \n",
       "NAA15                   0.620830  \n",
       "PTEN                    0.894637  \n",
       "POGZ                    0.953178  \n",
       "KMT2C                   0.908995  \n",
       "KDM5B                   0.698533  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ASD_CHD_df = pd.read_excel('../data/supplemental_tables_200604.xlsx',sheet_name='Table S1',skiprows=1)\n",
    "\n",
    "print(len(ASD_CHD_df))\n",
    "ASD_CHD_df.index=ASD_CHD_df['gene']\n",
    "ASD_CHD_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "44\n",
      "78\n",
      "98\n",
      "174\n"
     ]
    }
   ],
   "source": [
    "# where do shared phenotype dnvs fall within the ontology?\n",
    "shared_DNV_genes = ASD_CHD_df[ASD_CHD_df['PCGC/PHN dual phenotype dDNV']==1].index.tolist()\n",
    "print(len(shared_DNV_genes))\n",
    "\n",
    "decipher_SNV_genes = ASD_CHD_df[ASD_CHD_df['dual_phenotype_snv_indel_DECIPHERv9.25']>0].index.tolist()\n",
    "print(len(decipher_SNV_genes))\n",
    "\n",
    "decipher_CNV_genes = ASD_CHD_df[ASD_CHD_df['dual_phenotype_small_CNV_less1MB_DECIPHERv9.25']>0].index.tolist()\n",
    "print(len(decipher_CNV_genes))\n",
    "all_val_genes = list(np.unique(shared_DNV_genes+decipher_SNV_genes+decipher_CNV_genes))\n",
    "print(len(all_val_genes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load PCnet\n",
    "\n",
    "Download PCnet from https://ndexbio.org/#/network/f93f402c-86d4-11e7-a10d-0ac135e8bacf (UUID: f93f402c-86d4-11e7-a10d-0ac135e8bacf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19781\n",
      "2724724\n"
     ]
    }
   ],
   "source": [
    "G_pcnet = nx.read_gpickle('/Users/brinrosenthal/Documents/CCBB_tickets_data/PCnet/G_PCnet.gpickle')\n",
    "\n",
    "print(len(G_pcnet.nodes()))\n",
    "print(len(G_pcnet.edges()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "844\n",
      "28559\n"
     ]
    }
   ],
   "source": [
    "# make a subgraph from ASD_CHD_df genes\n",
    "G_ASD_CHD = nx.subgraph(G_pcnet,ASD_CHD_df.index.tolist())\n",
    "print(len(G_ASD_CHD.nodes()))\n",
    "print(len(G_ASD_CHD.edges()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load similarity matrix "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>node1</th>\n",
       "      <th>node2</th>\n",
       "      <th>sim</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMARCC2</td>\n",
       "      <td>HIST1H4B</td>\n",
       "      <td>0.981613</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMARCC2</td>\n",
       "      <td>ZCCHC11</td>\n",
       "      <td>0.760676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMARCC2</td>\n",
       "      <td>SMARCC1</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMARCC2</td>\n",
       "      <td>ZDHHC12</td>\n",
       "      <td>0.212930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMARCC2</td>\n",
       "      <td>HIST1H4J</td>\n",
       "      <td>0.983393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     node1     node2       sim\n",
       "0  SMARCC2  HIST1H4B  0.981613\n",
       "1  SMARCC2   ZCCHC11  0.760676\n",
       "2  SMARCC2   SMARCC1  1.000000\n",
       "3  SMARCC2   ZDHHC12  0.212930\n",
       "4  SMARCC2  HIST1H4J  0.983393"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sim_rank_EL = pd.read_csv('new_clixo_files/sim_rank_cosine_EL.tsv',sep='\\t',names=['node1','node2','sim'])\n",
    "sim_rank_EL.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read in the ontology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "alpha='01'\n",
    "beta='45'\n",
    "filepath='new_clixo_files/ASDCHD.cosine..'+alpha+'..'+beta+'.ont'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brinrosenthal/anaconda3/envs/ipykernel_py2/lib/python2.7/site-packages/ddot/Ontology.py:1606: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  table = pd.read_table(table, comment='#', header=header)\n"
     ]
    }
   ],
   "source": [
    "ont = Ontology.from_table('new_clixo_files/ASDCHD_parsed_alpha.cosine.'+alpha+'_beta_.'+beta+'.txt',parent=0,child=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "844 genes, 120 terms, 2270 gene-term relations, 133 term-term relations\n",
       "node_attributes: []\n",
       "edge_attributes: ['Gene-Term']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add the fraction validated genes to ontology terms, write out for input to cytoscape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "963    844\n",
       "962    344\n",
       "961    324\n",
       "960    302\n",
       "959    240\n",
       "dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "term_sizes = pd.Series(ont.term_sizes)\n",
    "term_sizes.index = pd.Series(ont.terms_index).index.tolist()\n",
    "term_sizes=term_sizes.sort_values(ascending=False)\n",
    "term_sizes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AC008575.1</th>\n",
       "      <th>AC067968.1</th>\n",
       "      <th>ACHE</th>\n",
       "      <th>ACTB</th>\n",
       "      <th>ACTN2</th>\n",
       "      <th>ACTN4</th>\n",
       "      <th>ADAM17</th>\n",
       "      <th>ADGRL2</th>\n",
       "      <th>ADNP</th>\n",
       "      <th>AFDN</th>\n",
       "      <th>...</th>\n",
       "      <th>954</th>\n",
       "      <th>955</th>\n",
       "      <th>956</th>\n",
       "      <th>957</th>\n",
       "      <th>958</th>\n",
       "      <th>959</th>\n",
       "      <th>960</th>\n",
       "      <th>961</th>\n",
       "      <th>962</th>\n",
       "      <th>963</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AC008575.1</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC067968.1</th>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACHE</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTB</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTN2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 964 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            AC008575.1  AC067968.1   ACHE   ACTB  ACTN2  ACTN4  ADAM17  \\\n",
       "AC008575.1        True       False  False  False  False  False   False   \n",
       "AC067968.1       False        True  False  False  False  False   False   \n",
       "ACHE             False       False   True  False  False  False   False   \n",
       "ACTB             False       False  False   True  False  False   False   \n",
       "ACTN2            False       False  False  False   True  False   False   \n",
       "\n",
       "            ADGRL2   ADNP   AFDN  ...    954    955    956    957    958  \\\n",
       "AC008575.1   False  False  False  ...  False   True  False  False   True   \n",
       "AC067968.1   False  False  False  ...  False  False  False  False  False   \n",
       "ACHE         False  False  False  ...  False  False  False  False   True   \n",
       "ACTB         False  False  False  ...   True  False  False   True  False   \n",
       "ACTN2        False  False  False  ...  False  False  False  False   True   \n",
       "\n",
       "              959    960    961    962   963  \n",
       "AC008575.1  False   True   True   True  True  \n",
       "AC067968.1  False  False  False  False  True  \n",
       "ACHE        False  False   True  False  True  \n",
       "ACTB        False   True  False   True  True  \n",
       "ACTN2       False   True   True  False  True  \n",
       "\n",
       "[5 rows x 964 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# need to use ont_connected to get actual genes in term (trace ontology)\n",
    "\n",
    "ont_connected = pd.DataFrame(ont.connected())\n",
    "ont_connected.index=pd.Series(ont.genes_index).index.tolist()+pd.Series(ont.terms_index).index.tolist()\n",
    "ont_connected.columns=pd.Series(ont.genes_index).index.tolist()+pd.Series(ont.terms_index).index.tolist()\n",
    "ont_connected.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>844</th>\n",
       "      <th>845</th>\n",
       "      <th>846</th>\n",
       "      <th>847</th>\n",
       "      <th>848</th>\n",
       "      <th>849</th>\n",
       "      <th>850</th>\n",
       "      <th>851</th>\n",
       "      <th>852</th>\n",
       "      <th>853</th>\n",
       "      <th>...</th>\n",
       "      <th>954</th>\n",
       "      <th>955</th>\n",
       "      <th>956</th>\n",
       "      <th>957</th>\n",
       "      <th>958</th>\n",
       "      <th>959</th>\n",
       "      <th>960</th>\n",
       "      <th>961</th>\n",
       "      <th>962</th>\n",
       "      <th>963</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AC008575.1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC067968.1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACHE</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTB</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTN2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 120 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              844    845    846    847    848    849    850    851    852  \\\n",
       "AC008575.1  False  False  False   True  False  False  False  False  False   \n",
       "AC067968.1  False  False  False  False  False  False  False  False  False   \n",
       "ACHE        False  False  False  False  False  False  False  False  False   \n",
       "ACTB        False  False  False  False  False  False  False  False  False   \n",
       "ACTN2       False  False  False  False  False  False  False  False   True   \n",
       "\n",
       "              853  ...    954    955    956    957    958    959    960  \\\n",
       "AC008575.1  False  ...  False   True  False  False   True  False   True   \n",
       "AC067968.1  False  ...  False  False  False  False  False  False  False   \n",
       "ACHE        False  ...  False  False  False  False   True  False  False   \n",
       "ACTB        False  ...   True  False  False   True  False  False   True   \n",
       "ACTN2       False  ...  False  False  False  False   True  False   True   \n",
       "\n",
       "              961    962   963  \n",
       "AC008575.1   True   True  True  \n",
       "AC067968.1  False  False  True  \n",
       "ACHE         True  False  True  \n",
       "ACTB        False   True  True  \n",
       "ACTN2        True  False  True  \n",
       "\n",
       "[5 rows x 120 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2term_full = ont_connected.iloc[0:len(ont.genes_index)].T.iloc[len(ont.genes_index):].T\n",
    "gene2term_full.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save out the pathways which have at least one validated xenopus gene\n",
    "xenopus_validated_genes=['CREBBP','KAT6A','KANSL1','NSD1','KMT2D',\n",
    "                        'ANKRD11','PTPN11','SCN2A']\n",
    "xenopus_pathways = gene2term_full.loc[xenopus_validated_genes]\n",
    "# xenopus_pathways.sum().sort_values(ascending=False).to_csv('new_clixo_files/xenopus_validated_pathways.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write out the gene2term matrix\n",
    "# gene2term_full.to_csv('new_clixo_files/gene2term_full_'+alpha+'_'+beta+'.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/brinrosenthal/anaconda3/envs/ipykernel_py2/lib/python2.7/site-packages/pandas/core/frame.py:6692: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  sort=sort)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_val_genes</th>\n",
       "      <th>cluster_size</th>\n",
       "      <th>num_DECIPHER_CNV</th>\n",
       "      <th>num_DECIPHER_SNV</th>\n",
       "      <th>num_shared_DNV</th>\n",
       "      <th>term</th>\n",
       "      <th>fractionVal</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>term</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>886</td>\n",
       "      <td>0.529412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>19</td>\n",
       "      <td>37</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>920</td>\n",
       "      <td>0.513514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>892</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>853</th>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>853</td>\n",
       "      <td>0.444444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852</th>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>852</td>\n",
       "      <td>0.444444</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     all_val_genes cluster_size num_DECIPHER_CNV num_DECIPHER_SNV  \\\n",
       "term                                                                \n",
       "886              9           17                4                4   \n",
       "920             19           37                5               13   \n",
       "892              9           18                3                7   \n",
       "853              4            9                2                4   \n",
       "852              4            9                2                2   \n",
       "\n",
       "     num_shared_DNV term fractionVal  \n",
       "term                                  \n",
       "886               3  886    0.529412  \n",
       "920               9  920    0.513514  \n",
       "892               3  892         0.5  \n",
       "853               2  853    0.444444  \n",
       "852               2  852    0.444444  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont2numVal = pd.DataFrame(columns=['term','num_shared_DNV','num_DECIPHER_SNV','num_DECIPHER_CNV','all_val_genes','cluster_size'])\n",
    "genes_index= pd.Series(ont.genes_index)\n",
    "\n",
    "for focal_term in ont.term_2_gene.keys():\n",
    "    focal_genes=gene2term_full[gene2term_full[focal_term]].index.tolist()     \n",
    "    \n",
    "    shared_DNV_temp = len(np.intersect1d(focal_genes,shared_DNV_genes))\n",
    "    DECIPHER_SNV_temp = len(np.intersect1d(focal_genes,decipher_SNV_genes))\n",
    "    DECIPHER_CNV_temp = len(np.intersect1d(focal_genes,decipher_CNV_genes))\n",
    "    allVal_temp = len(np.intersect1d(focal_genes,all_val_genes))\n",
    "    ont2numVal = ont2numVal.append(pd.DataFrame({'term':[focal_term],'num_shared_DNV':[shared_DNV_temp],\n",
    "                                                'num_DECIPHER_SNV':[DECIPHER_SNV_temp],'num_DECIPHER_CNV':[DECIPHER_CNV_temp],\n",
    "                                                'all_val_genes':[allVal_temp],'cluster_size':[len(focal_genes)]}),ignore_index=True)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "ont2numVal['fractionVal']=ont2numVal['all_val_genes']/ont2numVal['cluster_size']\n",
    "ont2numVal = ont2numVal.sort_values('fractionVal',ascending=False)\n",
    "ont2numVal.index=ont2numVal['term']\n",
    "ont2numVal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ont2numVal.to_csv('new_clixo_no_alignment/ont2numVal_cosine_alpha'+alpha+'_beta'+beta+'.txt',sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run enrichment on each cluster, after discussion with Trey\n",
    "\n",
    "Enrich each term separately, using gprofiler\n",
    "\n",
    "Also run GO alignment, but this happens in a separate notebook (clixo_GO_alignment/ASD_CHD_new_clixo_cosine_GO_alignment_200428.ipynb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gprofiler import GProfiler\n",
    "gp = GProfiler(\"MyToolName/0.1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>844</th>\n",
       "      <th>845</th>\n",
       "      <th>846</th>\n",
       "      <th>847</th>\n",
       "      <th>848</th>\n",
       "      <th>849</th>\n",
       "      <th>850</th>\n",
       "      <th>851</th>\n",
       "      <th>852</th>\n",
       "      <th>853</th>\n",
       "      <th>...</th>\n",
       "      <th>954</th>\n",
       "      <th>955</th>\n",
       "      <th>956</th>\n",
       "      <th>957</th>\n",
       "      <th>958</th>\n",
       "      <th>959</th>\n",
       "      <th>960</th>\n",
       "      <th>961</th>\n",
       "      <th>962</th>\n",
       "      <th>963</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AC008575.1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC067968.1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACHE</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTB</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTN2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 120 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              844    845    846    847    848    849    850    851    852  \\\n",
       "AC008575.1  False  False  False   True  False  False  False  False  False   \n",
       "AC067968.1  False  False  False  False  False  False  False  False  False   \n",
       "ACHE        False  False  False  False  False  False  False  False  False   \n",
       "ACTB        False  False  False  False  False  False  False  False  False   \n",
       "ACTN2       False  False  False  False  False  False  False  False   True   \n",
       "\n",
       "              853  ...    954    955    956    957    958    959    960  \\\n",
       "AC008575.1  False  ...  False   True  False  False   True  False   True   \n",
       "AC067968.1  False  ...  False  False  False  False  False  False  False   \n",
       "ACHE        False  ...  False  False  False  False   True  False  False   \n",
       "ACTB        False  ...   True  False  False   True  False  False   True   \n",
       "ACTN2       False  ...  False  False  False  False   True  False   True   \n",
       "\n",
       "              961    962   963  \n",
       "AC008575.1   True   True  True  \n",
       "AC067968.1  False  False  True  \n",
       "ACHE         True  False  True  \n",
       "ACTB        False   True  True  \n",
       "ACTN2        True  False  True  \n",
       "\n",
       "[5 rows x 120 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# loop over terms in gene2term_full\n",
    "gene2term_full.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "844\n",
      "6\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.820000e-13      0.357   0.833         14             5   \n",
      "1  8.960000e-13      0.172   1.000         29             5   \n",
      "2  9.260000e-13      0.238   0.833         21             5   \n",
      "3  1.020000e-12      0.217   0.833         23             5   \n",
      "4  4.100000e-12      0.114   1.000         44             5   \n",
      "\n",
      "              term.id                              term.name  \n",
      "0          GO:0001518   voltage-gated sodium channel complex  \n",
      "1   REAC:R-HSA-445095    Interaction between L1 and Ankyrins  \n",
      "2          GO:0034706                 sodium channel complex  \n",
      "3          GO:0005248  voltage-gated sodium channel activity  \n",
      "4  REAC:R-HSA-5576892         Phase 0 - rapid depolarisation  \n",
      "845\n",
      "7\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0   0.0271        0.2   0.333          5             1  REAC:R-HSA-5624138   \n",
      "\n",
      "                                           term.name  \n",
      "0  Trafficking of myristoylated proteins to the c...  \n",
      "846\n",
      "7\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  2.150000e-16      0.149     1.0         47             7  GO:0018024   \n",
      "1  4.200000e-16      0.113     1.0         62             7  GO:0016278   \n",
      "2  4.200000e-16      0.119     1.0         59             7  GO:0042054   \n",
      "3  4.200000e-16      0.115     1.0         61             7  GO:0016279   \n",
      "4  4.330000e-15      0.080     1.0         88             7  GO:0008276   \n",
      "\n",
      "                                     term.name  \n",
      "0  histone-lysine N-methyltransferase activity  \n",
      "1          lysine N-methyltransferase activity  \n",
      "2           histone methyltransferase activity  \n",
      "3  protein-lysine N-methyltransferase activity  \n",
      "4           protein methyltransferase activity  \n",
      "847\n",
      "8\n",
      "   p.value  precision  recall  term.size  overlap.size                term.id  \\\n",
      "0   0.0270      0.030   0.400         67             2  MIRNA:hsa-miR-6789-3p   \n",
      "1   0.0430      0.006   1.000        154             1             KEGG:04934   \n",
      "2   0.0443      0.024   0.333         83             2             GO:0008013   \n",
      "3   0.0443      0.006   0.500        510             3             GO:0005912   \n",
      "4   0.0443      0.006   0.500        527             3             GO:0070161   \n",
      "\n",
      "              term.name  \n",
      "0       hsa-miR-6789-3p  \n",
      "1      Cushing syndrome  \n",
      "2  beta-catenin binding  \n",
      "3     adherens junction  \n",
      "4    anchoring junction  \n",
      "848\n",
      "8\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0   0.0314      0.005     1.0        183             1  REAC:R-HSA-6811442   \n",
      "1   0.0314      0.024     1.0         42             1   REAC:R-HSA-983189   \n",
      "2   0.0314      0.006     1.0        154             1   REAC:R-HSA-983231   \n",
      "3   0.0314      0.009     1.0        115             1  REAC:R-HSA-8856688   \n",
      "4   0.0314      0.012     1.0         81             1  REAC:R-HSA-6811434   \n",
      "\n",
      "                                           term.name  \n",
      "0     Intra-Golgi and retrograde Golgi-to-ER traffic  \n",
      "1                                           Kinesins  \n",
      "2  Factors involved in megakaryocyte development ...  \n",
      "3                   Golgi-to-ER retrograde transport  \n",
      "4      COPI-dependent Golgi-to-ER retrograde traffic  \n",
      "849\n",
      "9\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000927      0.004   0.778       1640             7  GO:0003723   \n",
      "1  0.003560      0.028   0.333        109             3  GO:0071427   \n",
      "2  0.003560      0.023   0.333        128             3  GO:0071166   \n",
      "3  0.003560      0.024   0.333        127             3  GO:0071426   \n",
      "4  0.003560      0.028   0.333        109             3  GO:0006406   \n",
      "\n",
      "                                           term.name  \n",
      "0                                        RNA binding  \n",
      "1  mRNA-containing ribonucleoprotein complex expo...  \n",
      "2             ribonucleoprotein complex localization  \n",
      "3      ribonucleoprotein complex export from nucleus  \n",
      "4                           mRNA export from nucleus  \n",
      "850\n",
      "9\n",
      "    p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0  0.000002      0.667    1.00          3             2          CORUM:6527   \n",
      "1  0.002500      0.500    0.50          2             1          CORUM:6529   \n",
      "2  0.002500      0.333    0.50          3             1          CORUM:6528   \n",
      "3  0.034600      0.100    0.25         10             1   REAC:R-HSA-418886   \n",
      "4  0.034600      0.077    0.25         13             1  REAC:R-HSA-5140745   \n",
      "\n",
      "                                           term.name  \n",
      "0                          FLRT3-LPHN3-UNC5B complex  \n",
      "1                                FLRT3-LPHN3 complex  \n",
      "2                          FLRT3-LPHN3-UNC5D complex  \n",
      "3                  Netrin mediated repulsion signals  \n",
      "4  WNT5A-dependent internalization of FZD2, FZD5 ...  \n",
      "851\n",
      "9\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.380000e-15      0.500   0.667         12             6   \n",
      "1  1.380000e-15      0.226   0.778         31             7   \n",
      "2  3.190000e-14      0.121   0.778         58             7   \n",
      "3  3.190000e-14      0.121   0.778         58             7   \n",
      "4  3.190000e-14      0.273   0.667         22             6   \n",
      "\n",
      "              term.id                                          term.name  \n",
      "0  REAC:R-HSA-1980150                                Signaling by NOTCH4  \n",
      "1  REAC:R-HSA-2122948   Activated NOTCH1 Transmits Signal to the Nucleus  \n",
      "2  REAC:R-HSA-2894862  Constitutive Signaling by NOTCH1 HD+PEST Domai...  \n",
      "3  REAC:R-HSA-2894858  Signaling by NOTCH1 HD+PEST Domain Mutants in ...  \n",
      "4  REAC:R-HSA-2979096  NOTCH2 Activation and Transmission of Signal t...  \n",
      "852\n",
      "9\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.120000e-14      0.353   0.667         17             6   \n",
      "1  1.120000e-14      0.353   0.667         17             6   \n",
      "2  1.120000e-14      0.400   0.667         15             6   \n",
      "3  2.000000e-13      0.222   0.667         27             6   \n",
      "4  8.760000e-13      0.171   0.667         35             6   \n",
      "\n",
      "             term.id                                          term.name  \n",
      "0  REAC:R-HSA-442982  Ras activation upon Ca2+ influx through NMDA r...  \n",
      "1  REAC:R-HSA-438066  Unblocking of NMDA receptor, glutamate binding...  \n",
      "2  REAC:R-HSA-442729  CREB phosphorylation through the activation of...  \n",
      "3  REAC:R-HSA-442742  CREB phosphorylation through the activation of...  \n",
      "4  REAC:R-HSA-438064               Post NMDA receptor activation events  \n",
      "853\n",
      "9\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  3.450000e-10      0.012   1.000        768             9   \n",
      "1  6.730000e-09      0.008   1.000       1152             9   \n",
      "2  5.820000e-08      0.015   0.778        456             7   \n",
      "3  4.300000e-07      0.022   0.750        274             6   \n",
      "4  4.300000e-07      0.022   0.750        274             6   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2          GO:0016569  covalent chromatin modification  \n",
      "3  REAC:R-HSA-4839726           Chromatin organization  \n",
      "4  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "854\n",
      "10\n",
      "    p.value  precision  recall  term.size  overlap.size            term.id  \\\n",
      "0  0.000052      0.103   0.375         29             3  REAC:R-HSA-445095   \n",
      "1  0.000088      0.011   0.600        538             6         GO:0030424   \n",
      "2  0.000148      0.006   0.700       1155             7         GO:0043005   \n",
      "3  0.000331      0.024   0.400        167             4         GO:0035637   \n",
      "4  0.000378      0.005   0.700       1468             7         GO:0097458   \n",
      "\n",
      "                             term.name  \n",
      "0  Interaction between L1 and Ankyrins  \n",
      "1                                 axon  \n",
      "2                    neuron projection  \n",
      "3   multicellular organismal signaling  \n",
      "4                          neuron part  \n",
      "855\n",
      "10\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000058      0.007   0.778       1053             7  GO:0048666   \n",
      "1  0.000075      0.004   0.889       2012             8  GO:0048468   \n",
      "2  0.000079      0.097   0.333         31             3  GO:1902803   \n",
      "3  0.000079      0.005   0.778       1303             7  GO:0030182   \n",
      "4  0.000081      0.004   0.889       2281             8  GO:0007399   \n",
      "\n",
      "                                  term.name  \n",
      "0                        neuron development  \n",
      "1                          cell development  \n",
      "2  regulation of synaptic vesicle transport  \n",
      "3                    neuron differentiation  \n",
      "4                nervous system development  \n",
      "856\n",
      "10\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  4.030000e-11      0.185   0.625         27             5  GO:0032452   \n",
      "1  4.030000e-11      0.161   0.625         31             5  GO:0006482   \n",
      "2  4.030000e-11      0.161   0.625         31             5  GO:0008214   \n",
      "3  4.030000e-11      0.185   0.625         27             5  GO:0070076   \n",
      "4  4.030000e-11      0.172   0.625         29             5  GO:0016577   \n",
      "\n",
      "                      term.name  \n",
      "0  histone demethylase activity  \n",
      "1         protein demethylation  \n",
      "2          protein dealkylation  \n",
      "3  histone lysine demethylation  \n",
      "4         histone demethylation  \n",
      "857\n",
      "10\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0  0.00173      0.039   0.667         51             2  REAC:R-HSA-6802949   \n",
      "1  0.00183      0.027   0.667         74             2  REAC:R-HSA-6802957   \n",
      "2  0.01000      0.008   0.667        239             2  REAC:R-HSA-5673001   \n",
      "3  0.01000      0.008   0.667        245             2  REAC:R-HSA-5684996   \n",
      "4  0.01070      0.007   0.667        284             2  REAC:R-HSA-5683057   \n",
      "\n",
      "                        term.name  \n",
      "0        Signaling by RAS mutants  \n",
      "1        Oncogenic MAPK signaling  \n",
      "2          RAF/MAP kinase cascade  \n",
      "3           MAPK1/MAPK3 signaling  \n",
      "4  MAPK family signaling cascades  \n",
      "858\n",
      "10\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  5.880000e-28      0.370     1.0         27            10  GO:0019706   \n",
      "1  5.880000e-28      0.370     1.0         27            10  GO:0019707   \n",
      "2  2.060000e-27      0.323     1.0         31            10  GO:0016417   \n",
      "3  1.650000e-26      0.263     1.0         38            10  GO:0016409   \n",
      "4  3.510000e-24      0.321     0.9         28             9  GO:0018345   \n",
      "\n",
      "                                          term.name  \n",
      "0  protein-cysteine S-palmitoyltransferase activity  \n",
      "1       protein-cysteine S-acyltransferase activity  \n",
      "2                        S-acyltransferase activity  \n",
      "3                     palmitoyltransferase activity  \n",
      "4                            protein palmitoylation  \n",
      "859\n",
      "10\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.170000e-25      0.345     1.0         29            10   \n",
      "1  5.050000e-20      0.100     1.0        100            10   \n",
      "2  6.140000e-19      0.500     0.7         14             7   \n",
      "3  6.140000e-19      0.060     1.0        167            10   \n",
      "4  6.140000e-19      0.129     0.9         70             9   \n",
      "\n",
      "             term.id                             term.name  \n",
      "0  REAC:R-HSA-445095   Interaction between L1 and Ankyrins  \n",
      "1  REAC:R-HSA-373760                    L1CAM interactions  \n",
      "2         GO:0001518  voltage-gated sodium channel complex  \n",
      "3         GO:0035637    multicellular organismal signaling  \n",
      "4         GO:0019226         transmission of nerve impulse  \n",
      "860\n",
      "11\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.950000e-08      0.286   0.364         14             4   \n",
      "1  4.250000e-08      0.138   0.571         29             4   \n",
      "2  8.680000e-08      0.174   0.364         23             4   \n",
      "3  8.680000e-08      0.190   0.364         21             4   \n",
      "4  1.210000e-07      0.091   0.571         44             4   \n",
      "\n",
      "              term.id                              term.name  \n",
      "0          GO:0001518   voltage-gated sodium channel complex  \n",
      "1   REAC:R-HSA-445095    Interaction between L1 and Ankyrins  \n",
      "2          GO:0005248  voltage-gated sodium channel activity  \n",
      "3          GO:0034706                 sodium channel complex  \n",
      "4  REAC:R-HSA-5576892         Phase 0 - rapid depolarisation  \n",
      "861\n",
      "11\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000106      0.045   0.364         88             4  GO:0060996   \n",
      "1  0.000108      1.000   0.182          2             2  GO:0021816   \n",
      "2  0.000361      0.400   0.182          5             2  GO:0034334   \n",
      "3  0.000361      0.013   0.455        398             5  GO:0043547   \n",
      "4  0.000361      0.064   0.273         47             3  GO:0008347   \n",
      "\n",
      "                                           term.name  \n",
      "0                        dendritic spine development  \n",
      "1  extension of a leading process involved in cel...  \n",
      "2                      adherens junction maintenance  \n",
      "3             positive regulation of GTPase activity  \n",
      "4                               glial cell migration  \n",
      "862\n",
      "11\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.210000e-19      0.053   1.000        208            11  GO:0008168   \n",
      "1  1.210000e-19      0.050   1.000        218            11  GO:0016741   \n",
      "2  1.980000e-19      0.148   0.818         61             9  GO:0016279   \n",
      "3  1.980000e-19      0.145   0.818         62             9  GO:0016278   \n",
      "4  2.670000e-18      0.060   0.909        168            10  GO:0006479   \n",
      "\n",
      "                                           term.name  \n",
      "0                         methyltransferase activity  \n",
      "1  transferase activity, transferring one-carbon ...  \n",
      "2        protein-lysine N-methyltransferase activity  \n",
      "3                lysine N-methyltransferase activity  \n",
      "4                                protein methylation  \n",
      "863\n",
      "11\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000009      0.006   0.818       1468             9  GO:0097458   \n",
      "1  0.000024      0.007   0.727       1155             8  GO:0043005   \n",
      "2  0.000042      0.005   0.818       1980             9  GO:0042995   \n",
      "3  0.000046      0.008   0.636        853             7  GO:0045202   \n",
      "4  0.000148      0.016   0.455        305             5  GO:0050808   \n",
      "\n",
      "              term.name  \n",
      "0           neuron part  \n",
      "1     neuron projection  \n",
      "2       cell projection  \n",
      "3               synapse  \n",
      "4  synapse organization  \n",
      "864\n",
      "12\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000056      0.143    0.25         21             3  GO:0007413   \n",
      "1  0.000056      0.013    0.50        475             6  GO:0061564   \n",
      "2  0.000056      0.014    0.50        438             6  GO:0007409   \n",
      "3  0.000056      0.143    0.25         21             3  GO:0106030   \n",
      "4  0.000088      0.011    0.50        549             6  GO:0048667   \n",
      "\n",
      "                                           term.name  \n",
      "0                               axonal fasciculation  \n",
      "1                                   axon development  \n",
      "2                                       axonogenesis  \n",
      "3                    neuron projection fasciculation  \n",
      "4  cell morphogenesis involved in neuron differen...  \n",
      "865\n",
      "12\n",
      "    p.value  precision  recall  term.size  overlap.size            term.id  \\\n",
      "0  0.000002      0.667   1.000          3             2         CORUM:6527   \n",
      "1  0.000377      0.200   0.286         10             2  REAC:R-HSA-418889   \n",
      "2  0.000377      0.200   0.286         10             2  REAC:R-HSA-418886   \n",
      "3  0.000442      0.017   0.750        174             3         KEGG:04360   \n",
      "4  0.001690      0.018   0.333        226             4         GO:0007411   \n",
      "\n",
      "                                       term.name  \n",
      "0                      FLRT3-LPHN3-UNC5B complex  \n",
      "1  Ligand-independent caspase activation via DCC  \n",
      "2              Netrin mediated repulsion signals  \n",
      "3                                  Axon guidance  \n",
      "4                                  axon guidance  \n",
      "866\n",
      "12\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  4.080000e-10      0.105     0.5         57             6  GO:0043966   \n",
      "1  7.450000e-10      0.455     0.5         11             5  CORUM:2859   \n",
      "2  1.220000e-09      0.079     0.5         76             6  GO:0000123   \n",
      "3  1.310000e-09      0.070     0.5         86             6  GO:0031248   \n",
      "4  1.310000e-09      0.070     0.5         86             6  GO:1902493   \n",
      "\n",
      "                           term.name  \n",
      "0             histone H3 acetylation  \n",
      "1                       ING5 complex  \n",
      "2  histone acetyltransferase complex  \n",
      "3  protein acetyltransferase complex  \n",
      "4          acetyltransferase complex  \n",
      "867\n",
      "13\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0  0.00688      0.039     0.4         51             2  REAC:R-HSA-6802949   \n",
      "1  0.00726      0.027     0.4         74             2  REAC:R-HSA-6802957   \n",
      "2  0.03200      0.062     0.2         16             1    REAC:R-HSA-70221   \n",
      "3  0.03200      0.048     0.2         21             1   REAC:R-HSA-373753   \n",
      "4  0.03200      0.008     0.4        239             2  REAC:R-HSA-5673001   \n",
      "\n",
      "                             term.name  \n",
      "0             Signaling by RAS mutants  \n",
      "1             Oncogenic MAPK signaling  \n",
      "2  Glycogen breakdown (glycogenolysis)  \n",
      "3          Nephrin family interactions  \n",
      "4               RAF/MAP kinase cascade  \n",
      "868\n",
      "13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.00119      0.002   0.923       5740            12  GO:0006139   \n",
      "1  0.00119      0.002   0.923       5126            12  GO:0090304   \n",
      "2  0.00119      0.333   0.154          6             2  GO:0070087   \n",
      "3  0.00119      0.002   0.846       4633            11  GO:0016070   \n",
      "4  0.00119      0.006   0.538       1152             7  GO:0051276   \n",
      "\n",
      "                                          term.name  \n",
      "0  nucleobase-containing compound metabolic process  \n",
      "1                    nucleic acid metabolic process  \n",
      "2                      chromo shadow domain binding  \n",
      "3                             RNA metabolic process  \n",
      "4                           chromosome organization  \n",
      "869\n",
      "13\n",
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.00562      0.333   0.333          3             1  CORUM:2767   \n",
      "1  0.00562      0.500   0.333          2             1   CORUM:263   \n",
      "2  0.00562      0.333   0.333          3             1   CORUM:618   \n",
      "3  0.00562      0.333   0.333          3             1  CORUM:6192   \n",
      "4  0.00562      0.333   0.333          3             1   CORUM:972   \n",
      "\n",
      "                               term.name  \n",
      "0      RAD50-MRE11-NBN-p200-p350 complex  \n",
      "1      R/M complex (RAD50-MRE11 complex)  \n",
      "2  MRN complex (MRE11-RAD50-NBN complex)  \n",
      "3            CHDH-SQSTM1-MAP1LC3 complex  \n",
      "4  MRN complex (MRE11-RAD50-NBN complex)  \n",
      "870\n",
      "13\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  4.060000e-08      0.038   0.462        158             6  GO:0006475   \n",
      "1  4.060000e-08      0.041   0.462        148             6  GO:0016573   \n",
      "2  4.060000e-08      0.079   0.385         63             5  GO:0140034   \n",
      "3  4.060000e-08      0.012   0.692        768             9  GO:0006325   \n",
      "4  4.060000e-08      0.079   0.385         63             5  GO:0035064   \n",
      "\n",
      "                                 term.name  \n",
      "0  internal protein amino acid acetylation  \n",
      "1                      histone acetylation  \n",
      "2    methylation-dependent protein binding  \n",
      "3                   chromatin organization  \n",
      "4               methylated histone binding  \n",
      "871\n",
      "13\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  2.910000e-25      0.370   0.769         27            10  GO:0019707   \n",
      "1  2.910000e-25      0.370   0.769         27            10  GO:0019706   \n",
      "2  1.020000e-24      0.323   0.769         31            10  GO:0016417   \n",
      "3  8.130000e-24      0.263   0.769         38            10  GO:0016409   \n",
      "4  4.340000e-22      0.321   0.692         28             9  GO:0018345   \n",
      "\n",
      "                                          term.name  \n",
      "0       protein-cysteine S-acyltransferase activity  \n",
      "1  protein-cysteine S-palmitoyltransferase activity  \n",
      "2                        S-acyltransferase activity  \n",
      "3                     palmitoyltransferase activity  \n",
      "4                            protein palmitoylation  \n",
      "872\n",
      "14\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  9.610000e-15      0.017   0.929        768            13   \n",
      "1  9.480000e-13      0.011   0.929       1152            13   \n",
      "2  2.050000e-10      0.033   0.692        274             9   \n",
      "3  2.050000e-10      0.033   0.692        274             9   \n",
      "4  6.880000e-10      0.118   0.462         51             6   \n",
      "\n",
      "              term.id                    term.name  \n",
      "0          GO:0006325       chromatin organization  \n",
      "1          GO:0051276      chromosome organization  \n",
      "2  REAC:R-HSA-3247509  Chromatin modifying enzymes  \n",
      "3  REAC:R-HSA-4839726       Chromatin organization  \n",
      "4  REAC:R-HSA-3214842    HDMs demethylate histones  \n",
      "873\n",
      "14\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  5.180000e-30      0.132     1.0        106            14   \n",
      "1  7.660000e-30      0.123     1.0        114            14   \n",
      "2  1.490000e-29      0.192     1.0         73            14   \n",
      "3  9.120000e-29      0.163     1.0         86            14   \n",
      "4  1.210000e-28      0.156     1.0         90            14   \n",
      "\n",
      "              term.id                                          term.name  \n",
      "0          GO:0000786                                         nucleosome  \n",
      "1          GO:0044815                              DNA packaging complex  \n",
      "2  REAC:R-HSA-2299718               Condensation of Prophase Chromosomes  \n",
      "3   REAC:R-HSA-912446                              Meiotic recombination  \n",
      "4   REAC:R-HSA-201722  Formation of the beta-catenin:TCF transactivat...  \n",
      "874\n",
      "14\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.580000e-07      0.008   0.786       1441            11   \n",
      "1  1.580000e-07      0.007   0.786       1535            11   \n",
      "2  3.760000e-07      0.273   0.500         11             3   \n",
      "3  2.640000e-06      0.009   0.643       1051             9   \n",
      "4  5.490000e-06      0.005   0.786       2281            11   \n",
      "\n",
      "             term.id                         term.name  \n",
      "0         GO:0048699             generation of neurons  \n",
      "1         GO:0022008                      neurogenesis  \n",
      "2  REAC:R-HSA-376172                DSCAM interactions  \n",
      "3         GO:0032989  cellular component morphogenesis  \n",
      "4         GO:0007399        nervous system development  \n",
      "875\n",
      "14\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.820000e-14      0.500   0.500         12             6   \n",
      "1  2.820000e-14      0.226   0.583         31             7   \n",
      "2  7.380000e-13      0.121   0.583         58             7   \n",
      "3  7.380000e-13      0.121   0.583         58             7   \n",
      "4  7.380000e-13      0.273   0.500         22             6   \n",
      "\n",
      "              term.id                                          term.name  \n",
      "0  REAC:R-HSA-1980150                                Signaling by NOTCH4  \n",
      "1  REAC:R-HSA-2122948   Activated NOTCH1 Transmits Signal to the Nucleus  \n",
      "2  REAC:R-HSA-2644603                      Signaling by NOTCH1 in Cancer  \n",
      "3  REAC:R-HSA-2644602  Signaling by NOTCH1 PEST Domain Mutants in Cancer  \n",
      "4  REAC:R-HSA-2979096  NOTCH2 Activation and Transmission of Signal t...  \n",
      "876\n",
      "15\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  3.520000e-25      0.235   0.857         51            12   \n",
      "1  1.640000e-20      0.333   0.600         27             9   \n",
      "2  1.640000e-20      0.333   0.600         27             9   \n",
      "3  2.340000e-20      0.310   0.600         29             9   \n",
      "4  2.820000e-20      0.290   0.600         31             9   \n",
      "\n",
      "              term.id                     term.name  \n",
      "0  REAC:R-HSA-3214842     HDMs demethylate histones  \n",
      "1          GO:0070076  histone lysine demethylation  \n",
      "2          GO:0032452  histone demethylase activity  \n",
      "3          GO:0016577         histone demethylation  \n",
      "4          GO:0008214          protein dealkylation  \n",
      "877\n",
      "15\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  9.810000e-07      0.250   0.667         16             4  HP:0010819   \n",
      "1  1.400000e-05      0.114   0.667         35             4  HP:0002121   \n",
      "2  3.450000e-05      0.083   0.667         48             4  HP:0011146   \n",
      "3  2.530000e-04      0.048   0.667         84             4  HP:0002069   \n",
      "4  4.350000e-04      0.107   0.500         28             3  HP:0002373   \n",
      "\n",
      "                           term.name  \n",
      "0                    Atonic seizures  \n",
      "1                   Absence seizures  \n",
      "2                 Dialeptic seizures  \n",
      "3  Generalized tonic-clonic seizures  \n",
      "4                   Febrile seizures  \n",
      "878\n",
      "15\n",
      "   p.value  precision  recall  term.size  overlap.size            term.id  \\\n",
      "0  0.00125      0.333   1.000          3             1         CORUM:6528   \n",
      "1  0.00125      0.500   1.000          2             1         CORUM:6529   \n",
      "2  0.00125      0.333   1.000          3             1         CORUM:6527   \n",
      "3  0.02000      0.500   0.200          2             1  REAC:R-HSA-373756   \n",
      "4  0.03440      0.016   0.267        243             4         GO:0098742   \n",
      "\n",
      "                                           term.name  \n",
      "0                          FLRT3-LPHN3-UNC5D complex  \n",
      "1                                FLRT3-LPHN3 complex  \n",
      "2                          FLRT3-LPHN3-UNC5B complex  \n",
      "3                                   SDK interactions  \n",
      "4  cell-cell adhesion via plasma-membrane adhesio...  \n",
      "879\n",
      "15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0  0.000012      0.025   0.429        239             6  REAC:R-HSA-5673001   \n",
      "1  0.000012      0.027   0.429        221             6   REAC:R-HSA-112315   \n",
      "2  0.000012      0.024   0.429        245             6  REAC:R-HSA-5684996   \n",
      "3  0.000015      0.200   0.214         15             3   REAC:R-HSA-442729   \n",
      "4  0.000015      0.033   0.357        151             5   REAC:R-HSA-112314   \n",
      "\n",
      "                                           term.name  \n",
      "0                             RAF/MAP kinase cascade  \n",
      "1              Transmission across Chemical Synapses  \n",
      "2                              MAPK1/MAPK3 signaling  \n",
      "3  CREB phosphorylation through the activation of...  \n",
      "4  Neurotransmitter receptors and postsynaptic si...  \n",
      "880\n",
      "15\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  7.300000e-16      0.018   0.933        768            14   \n",
      "1  1.090000e-13      0.012   0.933       1152            14   \n",
      "2  2.710000e-10      0.042   0.533        192             8   \n",
      "3  4.900000e-10      0.033   0.643        274             9   \n",
      "4  4.900000e-10      0.033   0.643        274             9   \n",
      "\n",
      "              term.id                    term.name  \n",
      "0          GO:0006325       chromatin organization  \n",
      "1          GO:0051276      chromosome organization  \n",
      "2          GO:0042393              histone binding  \n",
      "3  REAC:R-HSA-4839726       Chromatin organization  \n",
      "4  REAC:R-HSA-3247509  Chromatin modifying enzymes  \n",
      "881\n",
      "16\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  4.940000e-12      0.033   0.818        274             9   \n",
      "1  4.940000e-12      0.033   0.818        274             9   \n",
      "2  9.950000e-12      0.047   0.562        192             9   \n",
      "3  6.430000e-11      0.092   0.438         76             7   \n",
      "4  6.840000e-11      0.081   0.438         86             7   \n",
      "\n",
      "              term.id                          term.name  \n",
      "0  REAC:R-HSA-3247509        Chromatin modifying enzymes  \n",
      "1  REAC:R-HSA-4839726             Chromatin organization  \n",
      "2          GO:0042393                    histone binding  \n",
      "3          GO:0000123  histone acetyltransferase complex  \n",
      "4          GO:1902493          acetyltransferase complex  \n",
      "882\n",
      "16\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000003      0.004   0.875       3435            14  GO:0005654   \n",
      "1  0.000003      0.003   0.938       4374            15  GO:0044428   \n",
      "2  0.000014      0.003   0.875       4026            14  GO:0031981   \n",
      "3  0.000154      0.003   0.875       5135            14  GO:0070013   \n",
      "4  0.000154      0.003   0.875       5135            14  GO:0031974   \n",
      "\n",
      "                       term.name  \n",
      "0                    nucleoplasm  \n",
      "1                   nuclear part  \n",
      "2                  nuclear lumen  \n",
      "3  intracellular organelle lumen  \n",
      "4        membrane-enclosed lumen  \n",
      "883\n",
      "16\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0   0.0314      0.006     1.0        154             1   REAC:R-HSA-983231   \n",
      "1   0.0314      0.012     1.0         81             1  REAC:R-HSA-6811434   \n",
      "2   0.0314      0.005     1.0        183             1  REAC:R-HSA-6811442   \n",
      "3   0.0314      0.024     1.0         42             1   REAC:R-HSA-983189   \n",
      "4   0.0314      0.009     1.0        115             1  REAC:R-HSA-8856688   \n",
      "\n",
      "                                           term.name  \n",
      "0  Factors involved in megakaryocyte development ...  \n",
      "1      COPI-dependent Golgi-to-ER retrograde traffic  \n",
      "2     Intra-Golgi and retrograde Golgi-to-ER traffic  \n",
      "3                                           Kinesins  \n",
      "4                   Golgi-to-ER retrograde transport  \n",
      "884\n",
      "16\n",
      "   p.value  precision  recall  term.size  overlap.size      term.id  \\\n",
      "0   0.0363      0.018   0.267        221             4  TF:M08406_0   \n",
      "1   0.0363      0.009   0.333        527             5    TF:M08444   \n",
      "2   0.0363      0.031   0.200         98             3    TF:M07082   \n",
      "3   0.0363      0.018   0.267        221             4    TF:M08406   \n",
      "4   0.0363      0.031   0.200         98             3  TF:M07082_0   \n",
      "\n",
      "                                           term.name  \n",
      "0  Factor: POU2F1:Sox-20; motif: WTGMATAACAAWR; m...  \n",
      "1         Factor: AP-4:PEA3; motif: NSCGGAWNCAGNTGNN  \n",
      "2                   Factor: DUX4; motif: TAAYYYAATCA  \n",
      "3        Factor: POU2F1:Sox-20; motif: WTGMATAACAAWR  \n",
      "4   Factor: DUX4; motif: TAAYYYAATCA; match class: 0  \n",
      "885\n",
      "17\n",
      "    p.value  precision  recall  term.size  overlap.size            term.id  \\\n",
      "0  0.000009      0.667   0.667          3             2         CORUM:6527   \n",
      "1  0.000199      0.011   0.600        530             6  REAC:R-HSA-422475   \n",
      "2  0.000417      0.060   0.300         50             3  REAC:R-HSA-373752   \n",
      "3  0.000641      0.200   0.200         10             2  REAC:R-HSA-418889   \n",
      "4  0.000641      0.200   0.200         10             2  REAC:R-HSA-418886   \n",
      "\n",
      "                                       term.name  \n",
      "0                      FLRT3-LPHN3-UNC5B complex  \n",
      "1                                  Axon guidance  \n",
      "2                             Netrin-1 signaling  \n",
      "3  Ligand-independent caspase activation via DCC  \n",
      "4              Netrin mediated repulsion signals  \n",
      "886\n",
      "17\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  8.600000e-19      0.040   0.824        348            14  GO:0000790   \n",
      "1  1.300000e-16      0.027   0.824        521            14  GO:0000785   \n",
      "2  1.330000e-16      0.026   0.824        537            14  GO:0044454   \n",
      "3  2.420000e-16      0.024   0.824        572            14  GO:0000228   \n",
      "4  6.380000e-15      0.015   0.882       1010            15  GO:0005694   \n",
      "\n",
      "                 term.name  \n",
      "0        nuclear chromatin  \n",
      "1                chromatin  \n",
      "2  nuclear chromosome part  \n",
      "3       nuclear chromosome  \n",
      "4               chromosome  \n",
      "887\n",
      "17\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  5.820000e-19      0.029   0.882        515            15  GO:0003682   \n",
      "1  6.540000e-19      0.021   0.941        768            16  GO:0006325   \n",
      "2  2.960000e-16      0.014   0.941       1152            16  GO:0051276   \n",
      "3  8.960000e-16      0.123   0.529         73             9  GO:0070603   \n",
      "4  2.980000e-15      0.025   0.765        521            13  GO:0000785   \n",
      "\n",
      "                          term.name  \n",
      "0                 chromatin binding  \n",
      "1            chromatin organization  \n",
      "2           chromosome organization  \n",
      "3  SWI/SNF superfamily-type complex  \n",
      "4                         chromatin  \n",
      "888\n",
      "17\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000051      0.017   0.412        410             7  GO:0003779   \n",
      "1  0.000197      0.009   0.471        879             8  GO:0008092   \n",
      "2  0.000197      0.056   0.235         72             4  GO:0016459   \n",
      "3  0.000254      0.008   0.471        956             8  GO:0000902   \n",
      "4  0.000416      0.008   0.471       1051             8  GO:0032989   \n",
      "\n",
      "                          term.name  \n",
      "0                     actin binding  \n",
      "1      cytoskeletal protein binding  \n",
      "2                    myosin complex  \n",
      "3                cell morphogenesis  \n",
      "4  cellular component morphogenesis  \n",
      "889\n",
      "17\n",
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.00337      0.333     0.5          3             1  CORUM:5461   \n",
      "1  0.00337      0.333     0.5          3             1  CORUM:3634   \n",
      "2  0.00337      0.333     0.5          3             1   CORUM:786   \n",
      "3  0.00337      0.500     0.5          2             1  CORUM:1042   \n",
      "4  0.00337      0.333     0.5          3             1  CORUM:6502   \n",
      "\n",
      "                                        term.name  \n",
      "0                 p50-p65 NF(kappa)B-SRC1 complex  \n",
      "1                        NR3C2-UBC9-SRC-1 complex  \n",
      "2                            MR-UBC9-SRC1 complex  \n",
      "3             SRA-SRC-1 ribonucleoprotein complex  \n",
      "4  Estrogen receptor complex (ESR1, EP300, NCOA1)  \n",
      "890\n",
      "17\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  7.720000e-07      0.006   0.812       2281            13   \n",
      "1  9.020000e-07      0.013   0.778        530             7   \n",
      "2  1.200000e-06      0.017   0.500        475             8   \n",
      "3  1.590000e-06      0.007   0.688       1535            11   \n",
      "4  2.340000e-06      0.667   1.000          3             2   \n",
      "\n",
      "             term.id                   term.name  \n",
      "0         GO:0007399  nervous system development  \n",
      "1  REAC:R-HSA-422475               Axon guidance  \n",
      "2         GO:0061564            axon development  \n",
      "3         GO:0022008                neurogenesis  \n",
      "4         CORUM:6527   FLRT3-LPHN3-UNC5B complex  \n",
      "891\n",
      "18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0   0.0147      0.009   0.333        549             5  GO:0048667   \n",
      "1   0.0147      0.008   0.333        624             5  GO:0048858   \n",
      "2   0.0147      0.008   0.333        643             5  GO:0032990   \n",
      "3   0.0147      0.008   0.333        622             5  GO:0120039   \n",
      "4   0.0147      0.008   0.333        608             5  GO:0048812   \n",
      "\n",
      "                                           term.name  \n",
      "0  cell morphogenesis involved in neuron differen...  \n",
      "1                      cell projection morphogenesis  \n",
      "2                            cell part morphogenesis  \n",
      "3  plasma membrane bounded cell projection morpho...  \n",
      "4                    neuron projection morphogenesis  \n",
      "892\n",
      "18\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  8.560000e-08      0.103   0.429         58             6  KEGG:05213   \n",
      "1  2.800000e-07      0.048   0.500        147             7  KEGG:05224   \n",
      "2  3.150000e-07      0.042   0.500        165             7  KEGG:05225   \n",
      "3  3.150000e-07      0.070   0.429         86             6  KEGG:05210   \n",
      "4  4.030000e-07      0.062   0.429         97             6  KEGG:05215   \n",
      "\n",
      "                  term.name  \n",
      "0        Endometrial cancer  \n",
      "1             Breast cancer  \n",
      "2  Hepatocellular carcinoma  \n",
      "3         Colorectal cancer  \n",
      "4           Prostate cancer  \n",
      "893\n",
      "18\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  7.730000e-12      0.037   0.750        245             9   \n",
      "1  7.730000e-12      0.038   0.750        239             9   \n",
      "2  1.970000e-11      0.032   0.750        284             9   \n",
      "3  9.370000e-10      0.081   0.500         74             6   \n",
      "4  1.440000e-08      0.098   0.417         51             5   \n",
      "\n",
      "              term.id                       term.name  \n",
      "0  REAC:R-HSA-5684996           MAPK1/MAPK3 signaling  \n",
      "1  REAC:R-HSA-5673001          RAF/MAP kinase cascade  \n",
      "2  REAC:R-HSA-5683057  MAPK family signaling cascades  \n",
      "3  REAC:R-HSA-6802957        Oncogenic MAPK signaling  \n",
      "4  REAC:R-HSA-6802949        Signaling by RAS mutants  \n",
      "894\n",
      "20\n",
      "895\n",
      "20\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.270000e-08      0.028    0.45        324             9  GO:0099177   \n",
      "1  1.270000e-08      0.028    0.45        323             9  GO:0050804   \n",
      "2  4.090000e-08      0.017    0.50        605            10  GO:0007268   \n",
      "3  4.090000e-08      0.017    0.50        605            10  GO:0098916   \n",
      "4  4.090000e-08      0.013    0.55        853            11  GO:0045202   \n",
      "\n",
      "                                      term.name  \n",
      "0        regulation of trans-synaptic signaling  \n",
      "1  modulation of chemical synaptic transmission  \n",
      "2                chemical synaptic transmission  \n",
      "3          anterograde trans-synaptic signaling  \n",
      "4                                       synapse  \n",
      "896\n",
      "21\n",
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.00234        0.2     1.0          5             1  CORUM:2318   \n",
      "1  0.00234        0.2     1.0          5             1  CORUM:2319   \n",
      "\n",
      "                          term.name  \n",
      "0  ITGA6-ITGB4-Laminin10/12 complex  \n",
      "1  ITGA6-ITGB4-Laminin10/12 complex  \n",
      "897\n",
      "21\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.850000e-36      0.517   0.833         29            15   \n",
      "1  2.970000e-27      0.150   0.833        100            15   \n",
      "2  1.070000e-21      0.171   0.571         70            12   \n",
      "3  1.070000e-21      0.084   0.667        167            14   \n",
      "4  1.020000e-20      0.095   0.619        137            13   \n",
      "\n",
      "             term.id                            term.name  \n",
      "0  REAC:R-HSA-445095  Interaction between L1 and Ankyrins  \n",
      "1  REAC:R-HSA-373760                   L1CAM interactions  \n",
      "2         GO:0019226        transmission of nerve impulse  \n",
      "3         GO:0035637   multicellular organismal signaling  \n",
      "4         GO:0001508                     action potential  \n",
      "898\n",
      "22\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.730000e-25      0.027   0.955        768            21   \n",
      "1  7.320000e-22      0.018   0.955       1152            21   \n",
      "2  3.020000e-15      0.047   0.684        274            13   \n",
      "3  3.020000e-15      0.047   0.684        274            13   \n",
      "4  2.230000e-13      0.029   0.591        456            13   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-4839726           Chromatin organization  \n",
      "3  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "899\n",
      "22\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  9.760000e-08      0.025   0.529        364             9   \n",
      "1  1.060000e-06      0.032   0.412        221             7   \n",
      "2  1.860000e-06      0.020   0.409        453             9   \n",
      "3  8.920000e-06      0.015   0.409        587             9   \n",
      "4  9.490000e-06      0.012   0.455        856            10   \n",
      "\n",
      "             term.id                                          term.name  \n",
      "0  REAC:R-HSA-112316                                    Neuronal System  \n",
      "1  REAC:R-HSA-112315              Transmission across Chemical Synapses  \n",
      "2         GO:0046873       metal ion transmembrane transporter activity  \n",
      "3         GO:0022890  inorganic cation transmembrane transporter act...  \n",
      "4         GO:0030001                                metal ion transport  \n",
      "900\n",
      "22\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0   0.0309      0.167   0.333          6             1  REAC:R-HSA-6782861   \n",
      "1   0.0322      0.043   0.333         23             1  REAC:R-HSA-2173795   \n",
      "2   0.0322      0.077   0.333         13             1  REAC:R-HSA-2892247   \n",
      "3   0.0322      0.023   0.333         44             1  REAC:R-HSA-2173793   \n",
      "4   0.0322      0.024   0.333         41             1  REAC:R-HSA-6782315   \n",
      "\n",
      "                                           term.name  \n",
      "0        Synthesis of wybutosine at G37 of tRNA(Phe)  \n",
      "1  Downregulation of SMAD2/3:SMAD4 transcriptiona...  \n",
      "2  POU5F1 (OCT4), SOX2, NANOG activate genes rela...  \n",
      "3  Transcriptional activity of SMAD2/SMAD3:SMAD4 ...  \n",
      "4       tRNA modification in the nucleus and cytosol  \n",
      "901\n",
      "22\n",
      "   p.value  precision  recall  term.size  overlap.size      term.id  \\\n",
      "0  0.00137      0.005   0.636       2837            14  TF:M07397_1   \n",
      "1  0.00326      0.004   0.682       3784            15  TF:M07226_1   \n",
      "2  0.00468      0.025   0.800        159             4   HP:0010720   \n",
      "3  0.00468      0.030   0.800        132             4   HP:0009553   \n",
      "4  0.00468      0.028   0.800        145             4   HP:0011361   \n",
      "\n",
      "                                           term.name  \n",
      "0  Factor: ZBP89; motif: CCCCKCCCCCNN; match clas...  \n",
      "1    Factor: SP1; motif: NCCCCKCCCCC; match class: 1  \n",
      "2                              Abnormal hair pattern  \n",
      "3                        Abnormality of the hairline  \n",
      "4                   Congenital abnormal hair pattern  \n",
      "902\n",
      "23\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.660000e-15      0.160   0.667         50             8   \n",
      "1  1.110000e-10      0.019   0.833        530            10   \n",
      "2  1.850000e-09      0.010   0.682       1441            15   \n",
      "3  2.310000e-09      0.010   0.682       1535            15   \n",
      "4  2.530000e-09      0.400   0.333         10             4   \n",
      "\n",
      "             term.id                          term.name  \n",
      "0  REAC:R-HSA-373752                 Netrin-1 signaling  \n",
      "1  REAC:R-HSA-422475                      Axon guidance  \n",
      "2         GO:0048699              generation of neurons  \n",
      "3         GO:0022008                       neurogenesis  \n",
      "4  REAC:R-HSA-418886  Netrin mediated repulsion signals  \n",
      "903\n",
      "23\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000020      0.028     0.3        212             6  GO:0014069   \n",
      "1  0.000020      0.028     0.3        213             6  GO:0099572   \n",
      "2  0.000020      0.027     0.3        221             6  GO:0098984   \n",
      "3  0.000020      0.028     0.3        217             6  GO:0032279   \n",
      "4  0.000023      0.025     0.3        236             6  GO:0045211   \n",
      "\n",
      "                     term.name  \n",
      "0         postsynaptic density  \n",
      "1  postsynaptic specialization  \n",
      "2     neuron to neuron synapse  \n",
      "3           asymmetric synapse  \n",
      "4        postsynaptic membrane  \n",
      "904\n",
      "23\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.090000e-17      0.055   0.706        218            12  GO:0016741   \n",
      "1  1.090000e-17      0.058   0.706        208            12  GO:0008168   \n",
      "2  5.070000e-17      0.065   0.647        168            11  GO:0006479   \n",
      "3  5.070000e-17      0.065   0.647        168            11  GO:0008213   \n",
      "4  7.760000e-17      0.148   0.529         61             9  GO:0016279   \n",
      "\n",
      "                                           term.name  \n",
      "0  transferase activity, transferring one-carbon ...  \n",
      "1                         methyltransferase activity  \n",
      "2                                protein methylation  \n",
      "3                                 protein alkylation  \n",
      "4        protein-lysine N-methyltransferase activity  \n",
      "905\n",
      "24\n",
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.00421      0.333   0.333          3             1  CORUM:3634   \n",
      "1  0.00421      0.333   0.333          3             1  CORUM:6502   \n",
      "2  0.00421      0.500   0.333          2             1  CORUM:2759   \n",
      "3  0.00421      0.500   0.333          2             1  CORUM:1042   \n",
      "4  0.00421      0.333   0.333          3             1  CORUM:2238   \n",
      "\n",
      "                                        term.name  \n",
      "0                        NR3C2-UBC9-SRC-1 complex  \n",
      "1  Estrogen receptor complex (ESR1, EP300, NCOA1)  \n",
      "2                               MBD1-MCAF complex  \n",
      "3             SRA-SRC-1 ribonucleoprotein complex  \n",
      "4                       MBD1-MCAF1-SETDB1 complex  \n",
      "906\n",
      "24\n",
      "   p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0  0.00187      0.333    1.00          3             1          CORUM:6848   \n",
      "1  0.00915      0.500    0.25          2             1   REAC:R-HSA-114294   \n",
      "2  0.00915      0.500    0.25          2             1   REAC:R-HSA-111452   \n",
      "3  0.01220      0.250    0.25          4             1    REAC:R-HSA-75108   \n",
      "4  0.01840      0.077    0.25         13             1  REAC:R-HSA-2892247   \n",
      "\n",
      "                                           term.name  \n",
      "0                    IRF2BP2-IRF2BP1-IRF2BPL complex  \n",
      "1  Activation, translocation and oligomerization ...  \n",
      "2      Activation and oligomerization of BAK protein  \n",
      "3  Activation, myristolyation of BID and transloc...  \n",
      "4  POU5F1 (OCT4), SOX2, NANOG activate genes rela...  \n",
      "907\n",
      "25\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  4.790000e-17      0.023    0.72        768            18  GO:0006325   \n",
      "1  3.270000e-14      0.016    0.72       1152            18  GO:0051276   \n",
      "2  8.140000e-13      0.066    0.40        152            10  GO:0018393   \n",
      "3  8.140000e-13      0.068    0.40        148            10  GO:0016573   \n",
      "4  9.150000e-13      0.063    0.40        158            10  GO:0006475   \n",
      "\n",
      "                                 term.name  \n",
      "0                   chromatin organization  \n",
      "1                  chromosome organization  \n",
      "2     internal peptidyl-lysine acetylation  \n",
      "3                      histone acetylation  \n",
      "4  internal protein amino acid acetylation  \n",
      "908\n",
      "26\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  3.040000e-08      0.016   0.522        768            12   \n",
      "1  9.060000e-08      0.011   0.565       1152            13   \n",
      "2  5.000000e-07      0.029   0.444        274             8   \n",
      "3  5.000000e-07      0.029   0.444        274             8   \n",
      "4  2.490000e-06      0.023   0.348        348             8   \n",
      "\n",
      "              term.id                    term.name  \n",
      "0          GO:0006325       chromatin organization  \n",
      "1          GO:0051276      chromosome organization  \n",
      "2  REAC:R-HSA-4839726       Chromatin organization  \n",
      "3  REAC:R-HSA-3247509  Chromatin modifying enzymes  \n",
      "4          GO:0000790            nuclear chromatin  \n",
      "909\n",
      "27\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000004      0.068   0.222         88             6  GO:0060996   \n",
      "1  0.000007      0.026   0.600        230             6  KEGG:04014   \n",
      "2  0.000015      0.044   0.222        136             6  GO:0050773   \n",
      "3  0.000015      0.076   0.185         66             5  GO:0060998   \n",
      "4  0.000015      0.032   0.259        220             7  GO:0016358   \n",
      "\n",
      "                                   term.name  \n",
      "0                dendritic spine development  \n",
      "1                      Ras signaling pathway  \n",
      "2         regulation of dendrite development  \n",
      "3  regulation of dendritic spine development  \n",
      "4                       dendrite development  \n",
      "910\n",
      "29\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  8.350000e-17      0.025   0.655        768            19  GO:0006325   \n",
      "1  8.080000e-14      0.016   0.655       1152            19  GO:0051276   \n",
      "2  5.550000e-12      0.066   0.345        152            10  GO:0018393   \n",
      "3  5.550000e-12      0.068   0.345        148            10  GO:0016573   \n",
      "4  6.230000e-12      0.062   0.345        160            10  GO:0018394   \n",
      "\n",
      "                              term.name  \n",
      "0                chromatin organization  \n",
      "1               chromosome organization  \n",
      "2  internal peptidyl-lysine acetylation  \n",
      "3                   histone acetylation  \n",
      "4           peptidyl-lysine acetylation  \n",
      "911\n",
      "29\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  6.070000e-39      0.311   0.655         61            19  GO:0016279   \n",
      "1  6.070000e-39      0.306   0.655         62            19  GO:0016278   \n",
      "2  1.560000e-37      0.131   0.759        168            22  GO:0006479   \n",
      "3  1.560000e-37      0.131   0.759        168            22  GO:0008213   \n",
      "4  1.570000e-37      0.160   0.724        131            21  GO:0016571   \n",
      "\n",
      "                                     term.name  \n",
      "0  protein-lysine N-methyltransferase activity  \n",
      "1          lysine N-methyltransferase activity  \n",
      "2                          protein methylation  \n",
      "3                           protein alkylation  \n",
      "4                          histone methylation  \n",
      "912\n",
      "32\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000018      0.005   0.719       4374            23  GO:0044428   \n",
      "1  0.000075      0.429   0.094          7             3  GO:1900246   \n",
      "2  0.000075      0.004   0.844       7066            27  GO:0005634   \n",
      "3  0.000143      0.006   0.594       3435            19  GO:0005654   \n",
      "4  0.000257      0.005   0.625       4026            20  GO:0031981   \n",
      "\n",
      "                                        term.name  \n",
      "0                                    nuclear part  \n",
      "1  positive regulation of RIG-I signaling pathway  \n",
      "2                                         nucleus  \n",
      "3                                     nucleoplasm  \n",
      "4                                   nuclear lumen  \n",
      "913\n",
      "32\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  3.670000e-13      0.258   0.320         31             8   \n",
      "1  6.840000e-13      0.188   0.391         48             9   \n",
      "2  2.340000e-12      0.318   0.280         22             7   \n",
      "3  2.360000e-12      0.545   0.240         11             6   \n",
      "4  3.540000e-12      0.500   0.240         12             6   \n",
      "\n",
      "              term.id                                          term.name  \n",
      "0  REAC:R-HSA-2122948   Activated NOTCH1 Transmits Signal to the Nucleus  \n",
      "1          KEGG:04330                            Notch signaling pathway  \n",
      "2  REAC:R-HSA-2979096  NOTCH2 Activation and Transmission of Signal t...  \n",
      "3   REAC:R-HSA-193692                    Regulated proteolysis of p75NTR  \n",
      "4  REAC:R-HSA-1980150                                Signaling by NOTCH4  \n",
      "914\n",
      "32\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.570000e-13      0.120   0.450         75             9   \n",
      "1  2.570000e-13      0.118   0.450         76             9   \n",
      "2  2.570000e-13      0.132   0.450         68             9   \n",
      "3  2.630000e-13      0.016   0.633       1152            19   \n",
      "4  2.630000e-13      0.022   0.567        768            17   \n",
      "\n",
      "              term.id                                          term.name  \n",
      "0  REAC:R-HSA-5693565  Recruitment and ATM-mediated phosphorylation o...  \n",
      "1  REAC:R-HSA-5693606                   DNA Double Strand Break Response  \n",
      "2  REAC:R-HSA-5693571                   Nonhomologous End-Joining (NHEJ)  \n",
      "3          GO:0051276                            chromosome organization  \n",
      "4          GO:0006325                             chromatin organization  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "915\n",
      "33\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  8.200000e-21      0.370     0.4         27            10  GO:0032452   \n",
      "1  8.200000e-21      0.370     0.4         27            10  GO:0070076   \n",
      "2  1.300000e-20      0.345     0.4         29            10  GO:0016577   \n",
      "3  1.720000e-20      0.323     0.4         31            10  GO:0006482   \n",
      "4  1.720000e-20      0.323     0.4         31            10  GO:0008214   \n",
      "\n",
      "                      term.name  \n",
      "0  histone demethylase activity  \n",
      "1  histone lysine demethylation  \n",
      "2         histone demethylation  \n",
      "3         protein demethylation  \n",
      "4          protein dealkylation  \n",
      "916\n",
      "33\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.400000e-15      0.057   0.609        245            14   \n",
      "1  5.660000e-15      0.049   0.609        284            14   \n",
      "2  2.090000e-14      0.054   0.565        239            13   \n",
      "3  1.800000e-10      0.018   0.500        869            16   \n",
      "4  1.800000e-10      0.019   0.500        852            16   \n",
      "\n",
      "              term.id                                       term.name  \n",
      "0  REAC:R-HSA-5684996                           MAPK1/MAPK3 signaling  \n",
      "1  REAC:R-HSA-5683057                  MAPK family signaling cascades  \n",
      "2  REAC:R-HSA-5673001                          RAF/MAP kinase cascade  \n",
      "3          GO:0023014  signal transduction by protein phosphorylation  \n",
      "4          GO:0000165                                    MAPK cascade  \n",
      "917\n",
      "33\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.080000e-12      0.021   0.545        869            18  GO:0023014   \n",
      "1  1.080000e-12      0.021   0.545        852            18  GO:0000165   \n",
      "2  2.180000e-11      0.011   0.667       1942            22  GO:0006928   \n",
      "3  4.310000e-11      0.009   0.727       2622            24  GO:0035556   \n",
      "4  1.990000e-10      0.021   0.455        709            15  GO:0043408   \n",
      "\n",
      "                                        term.name  \n",
      "0  signal transduction by protein phosphorylation  \n",
      "1                                    MAPK cascade  \n",
      "2       movement of cell or subcellular component  \n",
      "3               intracellular signal transduction  \n",
      "4                      regulation of MAPK cascade  \n",
      "918\n",
      "34\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  6.470000e-08      0.016   0.441        956            15  GO:0000902   \n",
      "1  6.470000e-08      0.005   0.882       5973            30  GO:0016043   \n",
      "2  6.560000e-08      0.005   0.882       6146            30  GO:0071840   \n",
      "3  7.030000e-08      0.008   0.618       2475            21  GO:0009653   \n",
      "4  9.830000e-08      0.014   0.441       1051            15  GO:0032989   \n",
      "\n",
      "                                       term.name  \n",
      "0                             cell morphogenesis  \n",
      "1                cellular component organization  \n",
      "2  cellular component organization or biogenesis  \n",
      "3             anatomical structure morphogenesis  \n",
      "4               cellular component morphogenesis  \n",
      "919\n",
      "36\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  5.420000e-33      0.039   0.833        768            30   \n",
      "1  5.540000e-28      0.026   0.833       1152            30   \n",
      "2  1.770000e-27      0.080   0.733        274            22   \n",
      "3  1.770000e-27      0.080   0.733        274            22   \n",
      "4  1.160000e-20      0.044   0.556        456            20   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "3  REAC:R-HSA-4839726           Chromatin organization  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "920\n",
      "37\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  5.480000e-13      0.172   0.357         58            10  KEGG:05213   \n",
      "1  5.480000e-13      0.079   0.464        165            13  KEGG:05225   \n",
      "2  1.650000e-11      0.116   0.357         86            10  KEGG:05210   \n",
      "3  2.150000e-11      0.085   0.393        129            11  KEGG:04068   \n",
      "4  6.120000e-11      0.075   0.393        147            11  KEGG:05224   \n",
      "\n",
      "                  term.name  \n",
      "0        Endometrial cancer  \n",
      "1  Hepatocellular carcinoma  \n",
      "2         Colorectal cancer  \n",
      "3    FoxO signaling pathway  \n",
      "4             Breast cancer  \n",
      "921\n",
      "38\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.130000e-47      0.048   0.974        768            37   \n",
      "1  4.590000e-41      0.032   0.974       1152            37   \n",
      "2  5.440000e-41      0.106   0.879        274            29   \n",
      "3  5.440000e-41      0.106   0.879        274            29   \n",
      "4  2.940000e-37      0.058   0.789        521            30   \n",
      "\n",
      "              term.id                    term.name  \n",
      "0          GO:0006325       chromatin organization  \n",
      "1          GO:0051276      chromosome organization  \n",
      "2  REAC:R-HSA-3247509  Chromatin modifying enzymes  \n",
      "3  REAC:R-HSA-4839726       Chromatin organization  \n",
      "4          GO:0000785                    chromatin  \n",
      "922\n",
      "38\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.190000e-19      0.370   0.333         27            10  GO:0032452   \n",
      "1  1.190000e-19      0.370   0.333         27            10  GO:0070076   \n",
      "2  1.880000e-19      0.345   0.333         29            10  GO:0016577   \n",
      "3  2.500000e-19      0.323   0.333         31            10  GO:0006482   \n",
      "4  2.500000e-19      0.323   0.333         31            10  GO:0008214   \n",
      "\n",
      "                      term.name  \n",
      "0  histone demethylase activity  \n",
      "1  histone lysine demethylation  \n",
      "2         histone demethylation  \n",
      "3         protein demethylation  \n",
      "4          protein dealkylation  \n",
      "923\n",
      "39\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.450000e-25      0.035   0.692        768            27  GO:0006325   \n",
      "1  3.720000e-21      0.023   0.692       1152            27  GO:0051276   \n",
      "2  5.470000e-17      0.049   0.436        346            17  GO:0018205   \n",
      "3  2.990000e-15      0.038   0.436        446            17  GO:0016570   \n",
      "4  3.470000e-15      0.037   0.436        456            17  GO:0016569   \n",
      "\n",
      "                         term.name  \n",
      "0           chromatin organization  \n",
      "1          chromosome organization  \n",
      "2     peptidyl-lysine modification  \n",
      "3             histone modification  \n",
      "4  covalent chromatin modification  \n",
      "924\n",
      "41\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  6.480000e-15      0.013   0.725       2281            29   \n",
      "1  3.600000e-12      0.015   0.575       1535            23   \n",
      "2  2.920000e-11      0.028   0.556        530            15   \n",
      "3  7.010000e-11      0.018   0.475       1053            19   \n",
      "4  9.560000e-11      0.015   0.525       1441            21   \n",
      "\n",
      "             term.id                   term.name  \n",
      "0         GO:0007399  nervous system development  \n",
      "1         GO:0022008                neurogenesis  \n",
      "2  REAC:R-HSA-422475               Axon guidance  \n",
      "3         GO:0048666          neuron development  \n",
      "4         GO:0048699       generation of neurons  \n",
      "925\n",
      "43\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.080000e-19      0.031   0.571        768            24   \n",
      "1  2.750000e-18      0.023   0.619       1152            26   \n",
      "2  2.540000e-16      0.055   0.600        274            15   \n",
      "3  2.540000e-16      0.055   0.600        274            15   \n",
      "4  4.830000e-16      0.177   0.262         62            11   \n",
      "\n",
      "              term.id                            term.name  \n",
      "0          GO:0006325               chromatin organization  \n",
      "1          GO:0051276              chromosome organization  \n",
      "2  REAC:R-HSA-4839726               Chromatin organization  \n",
      "3  REAC:R-HSA-3247509          Chromatin modifying enzymes  \n",
      "4          GO:0016278  lysine N-methyltransferase activity  \n",
      "926\n",
      "46\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  2.290000e-18      0.370   0.263         27            10  GO:0032452   \n",
      "1  2.290000e-18      0.370   0.263         27            10  GO:0070076   \n",
      "2  3.620000e-18      0.345   0.263         29            10  GO:0016577   \n",
      "3  4.790000e-18      0.323   0.263         31            10  GO:0008214   \n",
      "4  4.790000e-18      0.323   0.263         31            10  GO:0006482   \n",
      "\n",
      "                      term.name  \n",
      "0  histone demethylase activity  \n",
      "1  histone lysine demethylation  \n",
      "2         histone demethylation  \n",
      "3          protein dealkylation  \n",
      "4         protein demethylation  \n",
      "927\n",
      "47\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  7.320000e-12      0.258   0.235         31             8   \n",
      "1  1.310000e-11      0.188   0.300         48             9   \n",
      "2  2.160000e-11      0.318   0.206         22             7   \n",
      "3  2.160000e-11      0.545   0.176         11             6   \n",
      "4  3.240000e-11      0.500   0.176         12             6   \n",
      "\n",
      "              term.id                                          term.name  \n",
      "0  REAC:R-HSA-2122948   Activated NOTCH1 Transmits Signal to the Nucleus  \n",
      "1          KEGG:04330                            Notch signaling pathway  \n",
      "2  REAC:R-HSA-2979096  NOTCH2 Activation and Transmission of Signal t...  \n",
      "3   REAC:R-HSA-193692                    Regulated proteolysis of p75NTR  \n",
      "4  REAC:R-HSA-1980150                                Signaling by NOTCH4  \n",
      "928\n",
      "48\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  8.190000e-13      0.017   0.558       1441            24   \n",
      "1  1.710000e-12      0.016   0.558       1535            24   \n",
      "2  1.350000e-11      0.160   0.320         50             8   \n",
      "3  6.420000e-11      0.016   0.488       1303            21   \n",
      "4  6.420000e-11      0.011   0.605       2281            26   \n",
      "\n",
      "             term.id                   term.name  \n",
      "0         GO:0048699       generation of neurons  \n",
      "1         GO:0022008                neurogenesis  \n",
      "2  REAC:R-HSA-373752          Netrin-1 signaling  \n",
      "3         GO:0030182      neuron differentiation  \n",
      "4         GO:0007399  nervous system development  \n",
      "929\n",
      "48\n",
      "   p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.00297      0.010   0.333       1441            14  GO:0048699   \n",
      "1  0.00297      0.024   0.167        291             7  GO:0048562   \n",
      "2  0.00297      0.009   0.333       1535            14  GO:0022008   \n",
      "3  0.00552      0.066   0.095         61             4  GO:0003407   \n",
      "4  0.00832      0.200   0.250          5             1  CORUM:2319   \n",
      "\n",
      "                          term.name  \n",
      "0             generation of neurons  \n",
      "1     embryonic organ morphogenesis  \n",
      "2                      neurogenesis  \n",
      "3         neural retina development  \n",
      "4  ITGA6-ITGB4-Laminin10/12 complex  \n",
      "930\n",
      "48\n",
      "    p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  0.000188      0.018   0.250        549            10  GO:0048667   \n",
      "1  0.000188      0.019   0.225        475             9  GO:0061564   \n",
      "2  0.000188      0.016   0.250        608            10  GO:0048812   \n",
      "3  0.000188      0.016   0.250        622            10  GO:0120039   \n",
      "4  0.000188      0.016   0.250        624            10  GO:0048858   \n",
      "\n",
      "                                           term.name  \n",
      "0  cell morphogenesis involved in neuron differen...  \n",
      "1                                   axon development  \n",
      "2                    neuron projection morphogenesis  \n",
      "3  plasma membrane bounded cell projection morpho...  \n",
      "4                      cell projection morphogenesis  \n",
      "931\n",
      "49\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.620000e-15      0.014   0.673       2281            33  GO:0007399   \n",
      "1  4.780000e-15      0.036   0.408        549            20  GO:0048667   \n",
      "2  1.180000e-14      0.022   0.510       1155            25  GO:0043005   \n",
      "3  1.390000e-14      0.033   0.408        608            20  GO:0048812   \n",
      "4  1.390000e-14      0.018   0.551       1468            27  GO:0097458   \n",
      "\n",
      "                                           term.name  \n",
      "0                         nervous system development  \n",
      "1  cell morphogenesis involved in neuron differen...  \n",
      "2                                  neuron projection  \n",
      "3                    neuron projection morphogenesis  \n",
      "4                                        neuron part  \n",
      "932\n",
      "50\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.500000e-58      0.061   0.940        768            47   \n",
      "1  2.150000e-50      0.041   0.940       1152            47   \n",
      "2  3.240000e-44      0.124   0.773        274            34   \n",
      "3  3.240000e-44      0.124   0.773        274            34   \n",
      "4  2.590000e-42      0.069   0.720        521            36   \n",
      "\n",
      "              term.id                    term.name  \n",
      "0          GO:0006325       chromatin organization  \n",
      "1          GO:0051276      chromosome organization  \n",
      "2  REAC:R-HSA-4839726       Chromatin organization  \n",
      "3  REAC:R-HSA-3247509  Chromatin modifying enzymes  \n",
      "4          GO:0000785                    chromatin  \n",
      "933\n",
      "51\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  3.800000e-11      0.007   0.902       7066            46  GO:0005634   \n",
      "1  3.800000e-11      0.009   0.745       4374            38  GO:0044428   \n",
      "2  4.920000e-11      0.025   0.373        768            19  GO:0006325   \n",
      "3  4.920000e-11      0.010   0.667       3435            34  GO:0005654   \n",
      "4  6.170000e-10      0.009   0.686       4026            35  GO:0031981   \n",
      "\n",
      "                term.name  \n",
      "0                 nucleus  \n",
      "1            nuclear part  \n",
      "2  chromatin organization  \n",
      "3             nucleoplasm  \n",
      "4           nuclear lumen  \n",
      "934\n",
      "51\n",
      "    p.value  precision  recall  term.size  overlap.size             term.id  \\\n",
      "0  0.000028      0.429   0.125          7             3  REAC:R-HSA-2660826   \n",
      "1  0.000028      0.429   0.125          7             3  REAC:R-HSA-2660825   \n",
      "2  0.000028      0.068   0.208         73             5  REAC:R-HSA-1980143   \n",
      "3  0.000028      0.129   0.167         31             4  REAC:R-HSA-2122948   \n",
      "4  0.000079      0.250   0.125         12             3  REAC:R-HSA-1980150   \n",
      "\n",
      "                                           term.name  \n",
      "0  Constitutive Signaling by NOTCH1 t(7;9)(NOTCH1...  \n",
      "1  Signaling by NOTCH1 t(7;9)(NOTCH1:M1580_K2555)...  \n",
      "2                                Signaling by NOTCH1  \n",
      "3   Activated NOTCH1 Transmits Signal to the Nucleus  \n",
      "4                                Signaling by NOTCH4  \n",
      "935\n",
      "60\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  3.530000e-07      0.011   0.491       2475            28  GO:0009653   \n",
      "1  6.100000e-07      0.008   0.614       4150            35  GO:0048869   \n",
      "2  6.720000e-07      0.009   0.596       3980            34  GO:0030154   \n",
      "3  1.310000e-06      0.007   0.719       6001            41  GO:0032502   \n",
      "4  1.720000e-06      0.011   0.439       2281            25  GO:0007399   \n",
      "\n",
      "                            term.name  \n",
      "0  anatomical structure morphogenesis  \n",
      "1      cellular developmental process  \n",
      "2                cell differentiation  \n",
      "3               developmental process  \n",
      "4          nervous system development  \n",
      "936\n",
      "63\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  4.730000e-21      0.069   0.613        274            19   \n",
      "1  4.730000e-21      0.069   0.613        274            19   \n",
      "2  8.900000e-21      0.038   0.475        768            29   \n",
      "3  1.570000e-18      0.027   0.508       1152            31   \n",
      "4  5.700000e-16      0.194   0.197         62            12   \n",
      "\n",
      "              term.id                            term.name  \n",
      "0  REAC:R-HSA-4839726               Chromatin organization  \n",
      "1  REAC:R-HSA-3247509          Chromatin modifying enzymes  \n",
      "2          GO:0006325               chromatin organization  \n",
      "3          GO:0051276              chromosome organization  \n",
      "4          GO:0016278  lysine N-methyltransferase activity  \n",
      "937\n",
      "63\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  2.950000e-17      0.044   0.381        549            24  GO:0048667   \n",
      "1  5.030000e-17      0.017   0.619       2281            39  GO:0007399   \n",
      "2  7.940000e-17      0.039   0.381        608            24  GO:0048812   \n",
      "3  7.940000e-17      0.026   0.476       1155            30  GO:0043005   \n",
      "4  9.660000e-17      0.039   0.381        622            24  GO:0120039   \n",
      "\n",
      "                                           term.name  \n",
      "0  cell morphogenesis involved in neuron differen...  \n",
      "1                         nervous system development  \n",
      "2                    neuron projection morphogenesis  \n",
      "3                                  neuron projection  \n",
      "4  plasma membrane bounded cell projection morpho...  \n",
      "938\n",
      "65\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.590000e-21      0.255   0.448         51            13   \n",
      "1  1.210000e-18      0.407   0.193         27            11   \n",
      "2  1.210000e-18      0.407   0.193         27            11   \n",
      "3  2.130000e-18      0.379   0.193         29            11   \n",
      "4  3.120000e-18      0.355   0.193         31            11   \n",
      "\n",
      "              term.id                     term.name  \n",
      "0  REAC:R-HSA-3214842     HDMs demethylate histones  \n",
      "1          GO:0070076  histone lysine demethylation  \n",
      "2          GO:0032452  histone demethylase activity  \n",
      "3          GO:0016577         histone demethylation  \n",
      "4          GO:0008214          protein dealkylation  \n",
      "939\n",
      "66\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  2.470000e-10      0.012   0.606       3435            40   \n",
      "1  9.000000e-10      0.010   0.636       4091            42   \n",
      "2  9.000000e-10      0.012   0.545       2925            36   \n",
      "3  1.000000e-09      0.033   0.308        603            20   \n",
      "4  1.560000e-09      0.008   0.742       5816            49   \n",
      "\n",
      "               term.id                                          term.name  \n",
      "0           GO:0005654                                        nucleoplasm  \n",
      "1           GO:0019219  regulation of nucleobase-containing compound m...  \n",
      "2           GO:0051173  positive regulation of nitrogen compound metab...  \n",
      "3  MIRNA:hsa-miR-21-5p                                      hsa-miR-21-5p  \n",
      "4           GO:0051171  regulation of nitrogen compound metabolic process  \n",
      "940\n",
      "69\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.240000e-53      0.066   0.761        768            51   \n",
      "1  1.340000e-46      0.135   0.725        274            37   \n",
      "2  1.340000e-46      0.135   0.725        274            37   \n",
      "3  7.410000e-45      0.044   0.761       1152            51   \n",
      "4  1.590000e-37      0.079   0.537        456            36   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "2  REAC:R-HSA-4839726           Chromatin organization  \n",
      "3          GO:0051276          chromosome organization  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "941\n",
      "72\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.440000e-12      0.031   0.338        768            24  GO:0006325   \n",
      "1  7.670000e-12      0.011   0.676       4374            48  GO:0044428   \n",
      "2  7.670000e-12      0.013   0.606       3435            43  GO:0005654   \n",
      "3  1.180000e-11      0.008   0.831       7066            59  GO:0005634   \n",
      "4  2.150000e-10      0.022   0.352       1152            25  GO:0051276   \n",
      "\n",
      "                 term.name  \n",
      "0   chromatin organization  \n",
      "1             nuclear part  \n",
      "2              nucleoplasm  \n",
      "3                  nucleus  \n",
      "4  chromosome organization  \n",
      "942\n",
      "72\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.790000e-19      0.022   0.569       1852            41  GO:0006468   \n",
      "1  1.790000e-19      0.018   0.653       2622            47  GO:0035556   \n",
      "2  1.790000e-19      0.036   0.431        852            31  GO:0000165   \n",
      "3  2.410000e-19      0.036   0.431        869            31  GO:0023014   \n",
      "4  1.810000e-18      0.011   0.861       5833            62  GO:0007165   \n",
      "\n",
      "                                        term.name  \n",
      "0                         protein phosphorylation  \n",
      "1               intracellular signal transduction  \n",
      "2                                    MAPK cascade  \n",
      "3  signal transduction by protein phosphorylation  \n",
      "4                             signal transduction  \n",
      "943\n",
      "74\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  3.880000e-65      0.077   0.808        768            59   \n",
      "1  1.130000e-56      0.052   0.822       1152            60   \n",
      "2  3.050000e-42      0.131   0.655        274            36   \n",
      "3  3.050000e-42      0.131   0.655        274            36   \n",
      "4  3.770000e-37      0.081   0.507        456            37   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "3  REAC:R-HSA-4839726           Chromatin organization  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "944\n",
      "77\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  3.090000e-16      0.017   0.592       2622            45   \n",
      "1  3.090000e-16      0.034   0.382        852            29   \n",
      "2  3.480000e-16      0.033   0.382        869            29   \n",
      "3  7.020000e-14      0.073   0.321        245            18   \n",
      "4  3.210000e-13      0.063   0.321        284            18   \n",
      "\n",
      "              term.id                                       term.name  \n",
      "0          GO:0035556               intracellular signal transduction  \n",
      "1          GO:0000165                                    MAPK cascade  \n",
      "2          GO:0023014  signal transduction by protein phosphorylation  \n",
      "3  REAC:R-HSA-5684996                           MAPK1/MAPK3 signaling  \n",
      "4  REAC:R-HSA-5683057                  MAPK family signaling cascades  \n",
      "945\n",
      "82\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.080000e-12      0.017   0.526       2475            41  GO:0009653   \n",
      "1  1.590000e-12      0.022   0.410       1441            32  GO:0048699   \n",
      "2  2.160000e-12      0.019   0.462       1942            36  GO:0006928   \n",
      "3  2.600000e-12      0.010   0.756       5833            59  GO:0007165   \n",
      "4  3.830000e-12      0.021   0.410       1535            32  GO:0022008   \n",
      "\n",
      "                                   term.name  \n",
      "0         anatomical structure morphogenesis  \n",
      "1                      generation of neurons  \n",
      "2  movement of cell or subcellular component  \n",
      "3                        signal transduction  \n",
      "4                               neurogenesis  \n",
      "946\n",
      "92\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  8.290000e-26      0.552   0.219         29            16   \n",
      "1  8.850000e-21      0.190   0.260        100            19   \n",
      "2  9.940000e-21      0.035   0.440       1155            40   \n",
      "3  9.940000e-21      0.126   0.231        167            21   \n",
      "4  2.270000e-20      0.055   0.330        549            30   \n",
      "\n",
      "             term.id                                          term.name  \n",
      "0  REAC:R-HSA-445095                Interaction between L1 and Ankyrins  \n",
      "1  REAC:R-HSA-373760                                 L1CAM interactions  \n",
      "2         GO:0043005                                  neuron projection  \n",
      "3         GO:0035637                 multicellular organismal signaling  \n",
      "4         GO:0048667  cell morphogenesis involved in neuron differen...  \n",
      "947\n",
      "94\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  9.560000e-18      0.017   0.634       3435            59  GO:0005654   \n",
      "1  9.650000e-17      0.015   0.688       4374            64  GO:0044428   \n",
      "2  3.350000e-16      0.011   0.839       7066            78  GO:0005634   \n",
      "3  1.220000e-15      0.015   0.645       4026            60  GO:0031981   \n",
      "4  3.150000e-14      0.036   0.301        768            28  GO:0006325   \n",
      "\n",
      "                term.name  \n",
      "0             nucleoplasm  \n",
      "1            nuclear part  \n",
      "2                 nucleus  \n",
      "3           nuclear lumen  \n",
      "4  chromatin organization  \n",
      "948\n",
      "96\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.900000e-19      0.016   0.719       4374            69  GO:0044428   \n",
      "1  1.360000e-18      0.018   0.635       3435            61  GO:0005654   \n",
      "2  9.130000e-17      0.016   0.656       4026            63  GO:0031981   \n",
      "3  2.940000e-16      0.011   0.833       7066            80  GO:0005634   \n",
      "4  2.190000e-14      0.018   0.521       2752            50  GO:0009892   \n",
      "\n",
      "                                  term.name  \n",
      "0                              nuclear part  \n",
      "1                               nucleoplasm  \n",
      "2                             nuclear lumen  \n",
      "3                                   nucleus  \n",
      "4  negative regulation of metabolic process  \n",
      "949\n",
      "108\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  4.870000e-41      0.069   0.500        768            53   \n",
      "1  1.940000e-35      0.092   0.387        446            41   \n",
      "2  3.230000e-35      0.090   0.387        456            41   \n",
      "3  5.900000e-35      0.048   0.519       1152            55   \n",
      "4  7.370000e-34      0.124   0.493        274            34   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0016570             histone modification  \n",
      "2          GO:0016569  covalent chromatin modification  \n",
      "3          GO:0051276          chromosome organization  \n",
      "4  REAC:R-HSA-4839726           Chromatin organization  \n",
      "950\n",
      "109\n",
      "         p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  8.940000e-108      0.120   0.852        768            92   \n",
      "1   3.500000e-91      0.080   0.852       1152            92   \n",
      "2   4.950000e-81      0.237   0.691        274            65   \n",
      "3   4.950000e-81      0.237   0.691        274            65   \n",
      "4   1.760000e-68      0.136   0.574        456            62   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-4839726           Chromatin organization  \n",
      "3  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "4          GO:0016569  covalent chromatin modification  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "951\n",
      "110\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  4.610000e-25      0.052   0.392        768            40  GO:0006325   \n",
      "1  7.920000e-22      0.037   0.422       1152            43  GO:0051276   \n",
      "2  3.800000e-18      0.013   0.784       5973            80  GO:0016043   \n",
      "3  1.990000e-17      0.017   0.618       3634            63  GO:1903506   \n",
      "4  1.990000e-17      0.013   0.784       6146            80  GO:0071840   \n",
      "\n",
      "                                           term.name  \n",
      "0                             chromatin organization  \n",
      "1                            chromosome organization  \n",
      "2                    cellular component organization  \n",
      "3  regulation of nucleic acid-templated transcrip...  \n",
      "4      cellular component organization or biogenesis  \n",
      "952\n",
      "119\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.280000e-22      0.022   0.632       3435            74  GO:0005654   \n",
      "1  4.880000e-21      0.018   0.684       4374            80  GO:0044428   \n",
      "2  8.650000e-21      0.049   0.325        768            38  GO:0006325   \n",
      "3  1.980000e-20      0.019   0.650       4026            76  GO:0031981   \n",
      "4  1.720000e-19      0.037   0.368       1152            43  GO:0051276   \n",
      "\n",
      "                 term.name  \n",
      "0              nucleoplasm  \n",
      "1             nuclear part  \n",
      "2   chromatin organization  \n",
      "3            nuclear lumen  \n",
      "4  chromosome organization  \n",
      "953\n",
      "134\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  7.640000e-27      0.031   0.547       2281            70  GO:0007399   \n",
      "1  1.440000e-26      0.073   0.312        549            40  GO:0048667   \n",
      "2  1.690000e-25      0.082   0.281        438            36  GO:0007409   \n",
      "3  3.700000e-25      0.066   0.312        608            40  GO:0048812   \n",
      "4  4.690000e-25      0.037   0.445       1535            57  GO:0022008   \n",
      "\n",
      "                                           term.name  \n",
      "0                         nervous system development  \n",
      "1  cell morphogenesis involved in neuron differen...  \n",
      "2                                       axonogenesis  \n",
      "3                    neuron projection morphogenesis  \n",
      "4                                       neurogenesis  \n",
      "954\n",
      "144\n",
      "         p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  9.610000e-133      0.151   0.817        768           116   \n",
      "1  2.300000e-113      0.102   0.824       1152           117   \n",
      "2   1.500000e-95      0.281   0.675        274            77   \n",
      "3   1.500000e-95      0.281   0.675        274            77   \n",
      "4   3.490000e-83      0.169   0.542        456            77   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-4839726           Chromatin organization  \n",
      "3  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "955\n",
      "177\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  3.060000e-33      0.074   0.345        768            57  GO:0006325   \n",
      "1  5.110000e-27      0.052   0.364       1152            60  GO:0051276   \n",
      "2  9.040000e-23      0.026   0.570       3642            94  GO:2001141   \n",
      "3  9.040000e-23      0.026   0.570       3634            94  GO:1903506   \n",
      "4  2.020000e-22      0.026   0.564       3611            93  GO:0006355   \n",
      "\n",
      "                                           term.name  \n",
      "0                             chromatin organization  \n",
      "1                            chromosome organization  \n",
      "2             regulation of RNA biosynthetic process  \n",
      "3  regulation of nucleic acid-templated transcrip...  \n",
      "4         regulation of transcription, DNA-templated  \n",
      "956\n",
      "191\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  2.520000e-52      0.100   0.407        768            77  GO:0006325   \n",
      "1  4.690000e-47      0.073   0.444       1152            84  GO:0051276   \n",
      "2  2.980000e-44      0.037   0.667       3435           126  GO:0005654   \n",
      "3  1.490000e-39      0.031   0.709       4374           134  GO:0044428   \n",
      "4  2.270000e-39      0.032   0.683       4026           129  GO:0031981   \n",
      "\n",
      "                 term.name  \n",
      "0   chromatin organization  \n",
      "1  chromosome organization  \n",
      "2              nucleoplasm  \n",
      "3             nuclear part  \n",
      "4            nuclear lumen  \n",
      "957\n",
      "206\n",
      "         p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.010000e-135      0.174   0.684        768           134   \n",
      "1  2.700000e-114      0.118   0.694       1152           136   \n",
      "2   1.020000e-99      0.307   0.618        274            84   \n",
      "3   1.020000e-99      0.307   0.618        274            84   \n",
      "4   1.090000e-87      0.195   0.454        456            89   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-4839726           Chromatin organization  \n",
      "3  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "958\n",
      "220\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  5.640000e-35      0.045   0.495       2281           102  GO:0007399   \n",
      "1  6.750000e-32      0.053   0.398       1535            82  GO:0022008   \n",
      "2  2.140000e-31      0.055   0.383       1441            79  GO:0048699   \n",
      "3  3.790000e-30      0.089   0.262        608            54  GO:0048812   \n",
      "4  8.130000e-30      0.087   0.262        622            54  GO:0120039   \n",
      "\n",
      "                                           term.name  \n",
      "0                         nervous system development  \n",
      "1                                       neurogenesis  \n",
      "2                              generation of neurons  \n",
      "3                    neuron projection morphogenesis  \n",
      "4  plasma membrane bounded cell projection morpho...  \n",
      "959\n",
      "240\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  4.260000e-57      0.116   0.374        768            89  GO:0006325   \n",
      "1  1.090000e-51      0.045   0.647       3435           154  GO:0005654   \n",
      "2  1.220000e-50      0.084   0.408       1152            97  GO:0051276   \n",
      "3  7.450000e-47      0.039   0.668       4026           159  GO:0031981   \n",
      "4  3.700000e-46      0.037   0.689       4374           164  GO:0044428   \n",
      "\n",
      "                 term.name  \n",
      "0   chromatin organization  \n",
      "1              nucleoplasm  \n",
      "2  chromosome organization  \n",
      "3            nuclear lumen  \n",
      "4             nuclear part  \n",
      "960\n",
      "302\n",
      "         p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  1.890000e-110      0.180   0.486        768           138   \n",
      "1   1.020000e-90      0.123   0.500       1152           142   \n",
      "2   3.160000e-82      0.307   0.438        274            84   \n",
      "3   3.160000e-82      0.307   0.438        274            84   \n",
      "4   5.420000e-73      0.200   0.320        456            91   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1          GO:0051276          chromosome organization  \n",
      "2  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "3  REAC:R-HSA-4839726           Chromatin organization  \n",
      "4          GO:0016569  covalent chromatin modification  \n",
      "961\n",
      "324\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  1.810000e-31      0.053   0.407       2281           122  GO:0007399   \n",
      "1  3.060000e-30      0.066   0.317       1441            95  GO:0048699   \n",
      "2  3.060000e-30      0.064   0.327       1535            98  GO:0022008   \n",
      "3  1.890000e-28      0.068   0.293       1303            88  GO:0030182   \n",
      "4  9.020000e-27      0.099   0.200        608            60  GO:0048812   \n",
      "\n",
      "                         term.name  \n",
      "0       nervous system development  \n",
      "1            generation of neurons  \n",
      "2                     neurogenesis  \n",
      "3           neuron differentiation  \n",
      "4  neuron projection morphogenesis  \n",
      "962\n",
      "344\n",
      "        p.value  precision  recall  term.size  overlap.size     term.id  \\\n",
      "0  5.980000e-79      0.160   0.364        768           123  GO:0006325   \n",
      "1  4.150000e-67      0.115   0.391       1152           132  GO:0051276   \n",
      "2  6.340000e-67      0.061   0.621       3435           210  GO:0005654   \n",
      "3  8.950000e-62      0.054   0.648       4026           219  GO:0031981   \n",
      "4  8.950000e-62      0.193   0.260        456            88  GO:0016569   \n",
      "\n",
      "                         term.name  \n",
      "0           chromatin organization  \n",
      "1          chromosome organization  \n",
      "2                      nucleoplasm  \n",
      "3                    nuclear lumen  \n",
      "4  covalent chromatin modification  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "963\n",
      "844\n",
      "        p.value  precision  recall  term.size  overlap.size  \\\n",
      "0  7.350000e-77      0.228   0.225        768           175   \n",
      "1  6.470000e-65      0.358   0.226        274            98   \n",
      "2  6.470000e-65      0.358   0.226        274            98   \n",
      "3  3.050000e-60      0.268   0.157        456           122   \n",
      "4  3.830000e-59      0.164   0.243       1152           189   \n",
      "\n",
      "              term.id                        term.name  \n",
      "0          GO:0006325           chromatin organization  \n",
      "1  REAC:R-HSA-3247509      Chromatin modifying enzymes  \n",
      "2  REAC:R-HSA-4839726           Chromatin organization  \n",
      "3          GO:0016569  covalent chromatin modification  \n",
      "4          GO:0051276          chromosome organization  \n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "savedir_profile = 'gprofiler_clixo_ASD_CHD_cosine_alpha_'+alpha+'_beta_'+beta\n",
    "if not os.path.isdir(savedir_profile):\n",
    "    os.mkdir(savedir_profile)\n",
    "\n",
    "cluster2top_pathway = pd.DataFrame(columns=['cluster','pathway','term_ID',\n",
    "                                            'term_p-value','term_recall','term_precision','num_genes_in_cluster'])\n",
    "\n",
    "for focal_cluster in gene2term_full.columns.tolist():\n",
    "    print(focal_cluster)\n",
    "    focal_genes = gene2term_full[gene2term_full[focal_cluster]].index.tolist()\n",
    "    print(len(focal_genes))\n",
    "    if len(focal_genes)>5: # only run enrichment if > 5 genes in cluster\n",
    "        # ---- USE THIS FOR BACKGROUND OF all PCnet genes ----\n",
    "        gp_results = pd.DataFrame(gp.gprofile(focal_genes,custom_bg = G_pcnet.nodes(),correction_method=gp.THR_FDR))\n",
    "        \n",
    "        # ---- USE THIS FOR BACKGROUND OF 844 ASDCHD genes ----\n",
    "        # gp_results = pd.DataFrame(gp.gprofile(focal_genes,custom_bg = gene2term_full.index.tolist(),correction_method=gp.THR_FDR))\n",
    "        \n",
    "        if len(gp_results)>0: # make sure there are some significant enrichments\n",
    "\n",
    "            gp_results.columns = [\"query.number\", \"significant\", \"p.value\", \"term.size\",\n",
    "                                  \"query.size\", \"overlap.size\", \"recall\", \"precision\",\n",
    "                                  \"term.id\", \"domain\", \"subgraph.number\", \"term.name\",\n",
    "                                  \"relative.depth\", \"intersection\"]\n",
    "            print(gp_results[['p.value','precision','recall','term.size','overlap.size','term.id','term.name']].head())\n",
    "\n",
    "            # uncomment if we want to save the results\n",
    "            writer = pd.ExcelWriter(savedir_profile+'/cluster_'+str(focal_cluster)+'.xlsx')\n",
    "            gp_results.to_excel(writer)\n",
    "            writer.save()\n",
    "\n",
    "            cluster2top_pathway = cluster2top_pathway.append(pd.DataFrame({'cluster':[focal_cluster],\n",
    "                                                                          'pathway':gp_results.loc[0]['term.name'],\n",
    "                                                                           'term_ID':gp_results.loc[0]['term.id'],\n",
    "                                                                          'term_p-value':gp_results.loc[0]['p.value'],\n",
    "                                                                           'term_recall':gp_results.loc[0]['recall'],\n",
    "                                                                           'term_precision':gp_results.loc[0]['precision'],\n",
    "                                                                          'num_genes_in_cluster':[len(focal_genes)]}),\n",
    "                                                            ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            cluster2top_pathway = cluster2top_pathway.append(pd.DataFrame({'cluster':[focal_cluster],\n",
    "                                                                      'pathway':'None',\n",
    "                                                                       'term_ID':'None',\n",
    "                                                                      'term_p-value':1.0,\n",
    "                                                                        'term_recall':'None',\n",
    "                                                                        'term_precision':'None',\n",
    "                                                                      'num_genes_in_cluster':[len(focal_genes)]}),\n",
    "                                                        ignore_index=True)\n",
    "            \n",
    "    else:\n",
    "        cluster2top_pathway = cluster2top_pathway.append(pd.DataFrame({'cluster':[focal_cluster],\n",
    "                                                                      'pathway':'None',\n",
    "                                                                       'term_ID':'None',\n",
    "                                                                      'p-value':1.0,\n",
    "                                                                        'term_recall':'None',\n",
    "                                                                        'term_precision':'None',\n",
    "                                                                      'num_genes_in_cluster':[len(focal_genes)]}),\n",
    "                                                        ignore_index=True)\n",
    "\n",
    "#gp_results = pd.DataFrame(gp.gprofile(focal_genes,correction_method=gp.THR_FDR))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster</th>\n",
       "      <th>num_genes_in_cluster</th>\n",
       "      <th>pathway</th>\n",
       "      <th>term_ID</th>\n",
       "      <th>term_p-value</th>\n",
       "      <th>term_precision</th>\n",
       "      <th>term_recall</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cluster</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>844</th>\n",
       "      <td>844</td>\n",
       "      <td>6</td>\n",
       "      <td>voltage-gated sodium channel complex</td>\n",
       "      <td>GO:0001518</td>\n",
       "      <td>1.820000e-13</td>\n",
       "      <td>0.357</td>\n",
       "      <td>0.833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>845</th>\n",
       "      <td>845</td>\n",
       "      <td>7</td>\n",
       "      <td>Trafficking of myristoylated proteins to the c...</td>\n",
       "      <td>REAC:R-HSA-5624138</td>\n",
       "      <td>2.710000e-02</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>846</th>\n",
       "      <td>846</td>\n",
       "      <td>7</td>\n",
       "      <td>histone-lysine N-methyltransferase activity</td>\n",
       "      <td>GO:0018024</td>\n",
       "      <td>2.150000e-16</td>\n",
       "      <td>0.149</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>847</th>\n",
       "      <td>847</td>\n",
       "      <td>8</td>\n",
       "      <td>hsa-miR-6789-3p</td>\n",
       "      <td>MIRNA:hsa-miR-6789-3p</td>\n",
       "      <td>2.700000e-02</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>848</td>\n",
       "      <td>8</td>\n",
       "      <td>Intra-Golgi and retrograde Golgi-to-ER traffic</td>\n",
       "      <td>REAC:R-HSA-6811442</td>\n",
       "      <td>3.140000e-02</td>\n",
       "      <td>0.005</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        cluster num_genes_in_cluster  \\\n",
       "cluster                                \n",
       "844         844                    6   \n",
       "845         845                    7   \n",
       "846         846                    7   \n",
       "847         847                    8   \n",
       "848         848                    8   \n",
       "\n",
       "                                                   pathway  \\\n",
       "cluster                                                      \n",
       "844                   voltage-gated sodium channel complex   \n",
       "845      Trafficking of myristoylated proteins to the c...   \n",
       "846            histone-lysine N-methyltransferase activity   \n",
       "847                                        hsa-miR-6789-3p   \n",
       "848         Intra-Golgi and retrograde Golgi-to-ER traffic   \n",
       "\n",
       "                       term_ID  term_p-value term_precision term_recall  \n",
       "cluster                                                                  \n",
       "844                 GO:0001518  1.820000e-13          0.357       0.833  \n",
       "845         REAC:R-HSA-5624138  2.710000e-02            0.2       0.333  \n",
       "846                 GO:0018024  2.150000e-16          0.149           1  \n",
       "847      MIRNA:hsa-miR-6789-3p  2.700000e-02           0.03         0.4  \n",
       "848         REAC:R-HSA-6811442  3.140000e-02          0.005           1  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster2top_pathway.head()\n",
    "cluster2top_pathway.index=cluster2top_pathway['cluster']\n",
    "cluster2top_pathway.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# add top gprofiler term per cluster for further analysis/ visualization in cytoscape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add top pathway to ont2numVal\n",
    "ont2numVal=ont2numVal.join(cluster2top_pathway[['num_genes_in_cluster','term_p-value',\n",
    "                                                'term_precision','term_recall','pathway','term_ID']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "ont2numVal.to_csv('new_clixo_files/ont2numVal_cosine_alpha'+alpha+'_beta'+beta+'.txt',sep='\\t')\n",
    "# load precomputed ont2numVal so we don't have to enrich each pathway again\n",
    "# ont2numVal = pd.read_csv('new_clixo_files/ont2numVal_cosine_alpha'+alpha+'_beta'+beta+'.txt',sep='\\t',index_col='term')\n",
    "# ont2numVal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_val_genes</th>\n",
       "      <th>cluster_size</th>\n",
       "      <th>num_DECIPHER_CNV</th>\n",
       "      <th>num_DECIPHER_SNV</th>\n",
       "      <th>num_shared_DNV</th>\n",
       "      <th>term</th>\n",
       "      <th>fractionVal</th>\n",
       "      <th>num_genes_in_cluster</th>\n",
       "      <th>term_p-value</th>\n",
       "      <th>term_precision</th>\n",
       "      <th>term_recall</th>\n",
       "      <th>pathway</th>\n",
       "      <th>term_ID</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>term</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>886</td>\n",
       "      <td>0.529412</td>\n",
       "      <td>17</td>\n",
       "      <td>8.600000e-19</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.824</td>\n",
       "      <td>nuclear chromatin</td>\n",
       "      <td>GO:0000790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>19</td>\n",
       "      <td>37</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>920</td>\n",
       "      <td>0.513514</td>\n",
       "      <td>37</td>\n",
       "      <td>5.480000e-13</td>\n",
       "      <td>0.172</td>\n",
       "      <td>0.357</td>\n",
       "      <td>Endometrial cancer</td>\n",
       "      <td>KEGG:05213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>892</td>\n",
       "      <td>0.5</td>\n",
       "      <td>18</td>\n",
       "      <td>8.560000e-08</td>\n",
       "      <td>0.103</td>\n",
       "      <td>0.429</td>\n",
       "      <td>Endometrial cancer</td>\n",
       "      <td>KEGG:05213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>853</th>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>853</td>\n",
       "      <td>0.444444</td>\n",
       "      <td>9</td>\n",
       "      <td>3.450000e-10</td>\n",
       "      <td>0.012</td>\n",
       "      <td>1</td>\n",
       "      <td>chromatin organization</td>\n",
       "      <td>GO:0006325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852</th>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>852</td>\n",
       "      <td>0.444444</td>\n",
       "      <td>9</td>\n",
       "      <td>1.120000e-14</td>\n",
       "      <td>0.353</td>\n",
       "      <td>0.667</td>\n",
       "      <td>Ras activation upon Ca2+ influx through NMDA r...</td>\n",
       "      <td>REAC:R-HSA-442982</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     all_val_genes cluster_size num_DECIPHER_CNV num_DECIPHER_SNV  \\\n",
       "term                                                                \n",
       "886              9           17                4                4   \n",
       "920             19           37                5               13   \n",
       "892              9           18                3                7   \n",
       "853              4            9                2                4   \n",
       "852              4            9                2                2   \n",
       "\n",
       "     num_shared_DNV term fractionVal num_genes_in_cluster  term_p-value  \\\n",
       "term                                                                      \n",
       "886               3  886    0.529412                   17  8.600000e-19   \n",
       "920               9  920    0.513514                   37  5.480000e-13   \n",
       "892               3  892         0.5                   18  8.560000e-08   \n",
       "853               2  853    0.444444                    9  3.450000e-10   \n",
       "852               2  852    0.444444                    9  1.120000e-14   \n",
       "\n",
       "     term_precision term_recall  \\\n",
       "term                              \n",
       "886            0.04       0.824   \n",
       "920           0.172       0.357   \n",
       "892           0.103       0.429   \n",
       "853           0.012           1   \n",
       "852           0.353       0.667   \n",
       "\n",
       "                                                pathway            term_ID  \n",
       "term                                                                        \n",
       "886                                   nuclear chromatin         GO:0000790  \n",
       "920                                  Endometrial cancer         KEGG:05213  \n",
       "892                                  Endometrial cancer         KEGG:05213  \n",
       "853                              chromatin organization         GO:0006325  \n",
       "852   Ras activation upon Ca2+ influx through NMDA r...  REAC:R-HSA-442982  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont2numVal.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# write out ontology to cytoscape compatible format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# used to use the full ontology, but now we use the tree form for plotting\n",
    "# ont.to_table('../clixo_hierarchy/new_clixo_no_alignment/ASD_CHD_cosine_'+alpha+'_'+beta+'.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>child</th>\n",
       "      <th>parent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PLEKHA7</td>\n",
       "      <td>861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAM126A</td>\n",
       "      <td>963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ISX</td>\n",
       "      <td>874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SNTG2</td>\n",
       "      <td>953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PALM2</td>\n",
       "      <td>850</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     child parent\n",
       "0  PLEKHA7    861\n",
       "1  FAM126A    963\n",
       "2      ISX    874\n",
       "3    SNTG2    953\n",
       "4    PALM2    850"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# write out ontology in tree form (this is what is used in making the figures)\n",
    "tree = ont.get_tree()\n",
    "tree_df = pd.DataFrame(zip(*tree)).T\n",
    "tree_df.columns=['child','parent']\n",
    "tree_df.to_csv('new_clixo_files/tree_ASDCHD_cosine_'+alpha+'_'+beta+'.csv')\n",
    "tree_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
